\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Bayesian analysis}
\subtitle{Identifying essential genes by mutagenesis}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
  In this lecture, I'll present a case study on random transposon
  mutagenesis in Mycobacterium tuberculosis. Traditional frequentist
  methods are not suitable for this problem, and so I resorted to
  using Bayesian statistics, with Markov chain Monte Carlo.
}
} }



\begin{frame}{Mycobacterium tuberculosis}

  \bbi
\item The organism that causes tuberculosis.
  \bi
\item Cost for treatment: $\sim$\$15,000
\item Other bacterial pneumonias: $\sim$\$35
  \ei

\item 4.4 Mbp circular genome, completely sequenced
\item 4250 known or inferred genes
  \ei

\note{
  Tuberculosis can be surprisingly difficult and expensive to treat.
  So there's good reason to try to better understand its genome, to
  identify potential targets for new drugs.
}


\end{frame}




\begin{frame}[c]{}

  \centerline{\Large {\color{title} Goal}: identify the {\hilit essential} genes}

  \bigskip
  \bigskip
  \bigskip
  \bigskip

  \centerline{\Large {\color{title} Method}: random transposon mutagenesis}

\note{
  The current project seeks to identify the {\hilit essential} genes
  in the M.\ tuberculosis genome; the ones that if you knock them out,
  you get a non-viable mutant.

  This is done by random transposon mutageneis. A transposon is a bit
  of DNA that likes to insert itself into other DNA. So we'll randomly
  disrupt genes to find out which ones matter and which ones don't so much.
}

\end{frame}


\begin{frame}{\emph{Himar1\/} transposon}


\centerline{\includegraphics[width=0.7\textwidth]{Figs/himar.pdf}}

\vspace{-17mm}
\begin{center}
{\tt
\hspace*{2.5mm} 5'-TCGAAGCCTGCGAC{\vhilit
\textbf{TA}}ACGTT{\vhilit \textbf{TA}}AAGTTTG-3'

\hspace*{2.5mm} 3'-AGCTTCGGACGCTG{\vhilit
\textbf{AT}}TGCAA{\vhilit \textbf{AT}}TTCAAAC-5'
}
\end{center}

\hfill

\centerline{\hilit Note: $\ge$ 30 stop codons in each reading frame}

\note{
   They used the \emph{Himar1\/} transposon which randomly inserts
   itself into a site {\hilit TA}.

   ``Stop codons'' are codes for where a gene stops. The point is that
   if this thing gets inserted into a gene, it will end up with a
   truncated product.
  }
\end{frame}





\begin{frame}[c]{Sequence of the gene MT598}

\figw{Figs/mt598.pdf}{1.0}

\note{
}

\end{frame}


\begin{frame}[c]{Random transposon mutagenesis}

\only<1|handout 0>{\figh{Figs/mut1.pdf}{0.9}}
\only<2|handout 0>{\figh{Figs/mut2.pdf}{0.9}}
\only<3|handout 0>{\figh{Figs/mut3.pdf}{0.9}}
\only<4|handout 0>{\figh{Figs/mut4.pdf}{0.9}}
\only<5|handout 0>{\figh{Figs/mut5.pdf}{0.9}}
\only<6>{\figh{Figs/mut6.pdf}{0.9}}

\note{
}

\end{frame}


\begin{frame}{Random transposon mutagenesis}

  \bbi
  \item Location of transposon insertion determined by sequencing across
junctions

\item Viable insertion within a gene $\implies$ gene is non-essential

\item Essential genes: we will never see a viable insertion

\item {\hilit Complication}: Insertions in the very distal portion of
  an essential gene may not be sufficiently disruptive.

Thus, we omit from consideration insertions sites within the last 20\%
and last 100 bp of a gene.
\ei

\note{
}
\end{frame}






\begin{frame}{The data}

  \bbi
\item Number, locations of genes
\item Number of insertion sites in each gene
\item {\hilit $n$} viable mutants with exactly one transposon
    insertion
\item Location of the transposon insertion in each mutant
\ei

\note{
}
\end{frame}



\begin{frame}{TA sites in M. tuberculosis}

  \figh{Figs/numTAs.pdf}{0.4}

  \bbi
\item 74,403 sites
\item 65,649 sites within a gene
\item 57,934 sites within proximal portion of a gene
\item 4204/4250 genes with at least one TA site
  \ei

\note{
}
\end{frame}


\begin{frame}{1425 insertion mutants}

  \begin{columns}

    \column{0.4\textwidth}

      \vspace{-10mm}
      \figw{Figs/circlefig.pdf}{1.0}

    \column{0.6\textwidth}

    \bi
    \item 1425 insertion mutants
    \item 1025 within proximal portion of a gene
    \item 21 double-hits
    \item 770 unique genes hit
    \ei
  \end{columns}

\only<2->{
\hfill \begin{minipage}{0.9\textwidth}
  \bbi
  \item[\color{title} Questions:] {\hilit Proportion of essential genes in M.\ tuberculosis?}
  \item[] {\hilit Which genes are likely essential?}
  \ei
\end{minipage}
}

\note{
}
\end{frame}



\begin{frame}{Model}

  {\color{title} Transposon inserts completely at random}

  \bigskip

  \bi
\item Each TA site equally likely
\item Genes are either completely essential or completely
  non-essential
  \ei


  \note{
  }
  \end{frame}


\begin{frame}{Model}

\begin{columns}

\column{0.15\textwidth}

N genes \\[12pt]

n mutants

\column{0.85\textwidth}

$x_i$ = no. TA sites in gene $i$ \\[12pt]

$y_i$ = no. mutants with insertion in gene $i$.
\end{columns}


\bigskip\bigskip\bigskip

  $$  \theta_i = \left\{ \begin{array}{c} 1 \\ 0 \end{array} \text{ if gene $i$ is  }
      \begin{array}{c} \text{non-essential}  \\ \text{essential} \end{array} \right. $$


      \bigskip\bigskip\bigskip

      {\color{title} Model}: $\boldsymbol{y} \sim
      \text{multinomial}(n,\boldsymbol{p})$ \qquad where $p_i = x_i  \theta_i / \sum_j x_j \theta_j$

\bigskip\bigskip

      {\color{title} Goal}: Estimate $\theta_+ = \sum_i \theta_i$
      \qquad or \qquad $1 - \theta_+/N$

  \note{
  }
\end{frame}



\begin{frame}{The likelihood}


  \begin{eqnarray*}
    L(\boldsymbol{\theta} \ | \ \boldsymbol{y}) & = & {n \choose y} \prod_i (x_i \theta_i)^{y_i} / \sum_j (x_j \theta_j)^n \\[14pt]
                     & \propto & \left\{ \begin{array}{cl} (\sum_i x_i  \theta_i)^{-n} & \text{if $\theta_i$ = 1 whenever $y_i >$ 0} \\[12pt] 0  & \text{otherwise} \end{array}\right.
  \end{eqnarray*}



  \bigskip \bigskip

  {\color{title} Notes:}
  \bi
  \item Depends only on which $y_i>0$ and not on the specific values
  \item The MLE is $\hat{\theta}_i = 1\{y_i > 0\}$
    \ei


\note{
}
\end{frame}




\begin{frame}{The prior}

  \bigskip

  $\theta_+ \sim$ uniform on \{ 0, 1, ..., N \}

  \bigskip

$\boldsymbol{\theta} \ | \ \theta_+ \sim$  uniform over all sequences of 0's and 1's with $\theta_+$ 1's


  \bigskip \bigskip

  {\color{title} Notes:}
  \bi
\item We are assuming that $\text{Pr}(\theta_i = 1) = 1/2$
\item This is quite different from taking $\theta_i$ iid
  Bernoulli(1/2)
\item We are assuming that $\theta_i$ is independent of $x_i$ and the
  length of the gene
\item We could make use of information about the essential status of
  particular genes (e.g. known viable knock-outs)
  \ei


\note{
}

\end{frame}



\begin{frame}{A Gibbs sampler}

\bigskip

  {\color{title} Goal}: Estimate $\text{Pr}(\boldsymbol{\theta} \ | \ \boldsymbol{y})$

\bigskip

{\color{title} Gibbs sampler:}
\bi
\item Begin with some initial assignment $\boldsymbol{\theta}^{(0)}$
\item For iteration $s$, consider each gene one at a time
 \bi
  \item Let $\boldsymbol{\theta}_{-i}^{(s)} = (\theta_1^{(s+1)}, ..., \theta_{i-1}^{(s+1)}, \theta_{i+1}^{(s)}, ..., \theta_N^{(s)})$
  \item Calculate $\text{Pr}(\theta_i = 1 \ | \ \boldsymbol{\theta}_{-i}^{(s)}, \boldsymbol{y})$
  \item Assign $\theta_i^{(s)} = 1$ at random with that probability
 \ei
\item Repeat many times
  \ei

\bigskip \bigskip

This is an example of {\vhilit Markov chain Monte Carlo (MCMC)}.

  \note{
  }
\end{frame}


\begin{frame}[c]{MCMC in action}

\only<1|handout 0>{\figw{Figs/mcmc1.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/mcmc2.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/mcmc3.pdf}{1.0}}
\only<4|handout 0>{\figw{Figs/mcmc4.pdf}{1.0}}
\only<5|handout 0>{\figw{Figs/mcmc5.pdf}{1.0}}
\only<6|handout 0>{\figw{Figs/mcmc6.pdf}{1.0}}
\only<7|handout 0>{\figw{Figs/mcmc7.pdf}{1.0}}
\only<8|handout 0>{\figw{Figs/mcmc8.pdf}{1.0}}
\only<9|handout 0>{\figw{Figs/mcmc9.pdf}{1.0}}
\only<10|handout 0>{\figw{Figs/mcmc10.pdf}{1.0}}
\only<11|handout 0>{\figw{Figs/mcmc11.pdf}{1.0}}
\only<12|handout 0>{\figw{Figs/mcmc12.pdf}{1.0}}
\only<13>{\figw{Figs/mcmc13.pdf}{1.0}}

\note{
}
\end{frame}


\end{document}
