\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Bayesian analysis}
\subtitle{Identifying essential genes by mutagenesis}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
  In this lecture, I'll present a case study on random transposon
  mutagenesis in Mycobacterium tuberculosis. Traditional frequentist
  methods are not suitable for this problem, and so I resorted to
  using Bayesian statistics, with Markov chain Monte Carlo.
}
} }



\begin{frame}{Mycobacterium tuberculosis}

  \bbi
\item The organism that causes tuberculosis.
  \bi
\item Cost for treatment: $\sim$\$15,000
\item Other bacterial pneumonias: $\sim$\$35
  \ei

\item 4.4 Mbp circular genome, completely sequenced
\item 4250 known or inferred genes
  \ei

\note{
  Tuberculosis can be surprisingly difficult and expensive to treat.
  So there's good reason to try to better understand its genome, to
  identify potential targets for new drugs.
}


\end{frame}




\begin{frame}[c]{}

  \centerline{\Large {\color{title} Goal}: identify the {\hilit essential} genes}

  \bigskip
  \bigskip
  \bigskip
  \bigskip

  \centerline{\Large {\color{title} Method}: random transposon mutagenesis}

\note{
  The current project seeks to identify the {\hilit essential} genes
  in the M.\ tuberculosis genome; the ones that if you knock them out,
  you get a non-viable mutant.

  This is done by random transposon mutageneis. A transposon is a bit
  of DNA that likes to insert itself into other DNA. So we'll randomly
  disrupt genes to find out which ones matter and which ones don't so much.
}

\end{frame}


\begin{frame}{\emph{Himar1\/} transposon}


\centerline{\includegraphics[width=0.7\textwidth]{Figs/himar.pdf}}

\vspace{-17mm}
\begin{center}
{\tt
\hspace*{2.5mm} 5'-TCGAAGCCTGCGAC{\vhilit
\textbf{TA}}ACGTT{\vhilit \textbf{TA}}AAGTTTG-3'

\hspace*{2.5mm} 3'-AGCTTCGGACGCTG{\vhilit
\textbf{AT}}TGCAA{\vhilit \textbf{AT}}TTCAAAC-5'
}
\end{center}

\hfill

\centerline{\hilit Note: $\ge$ 30 stop codons in each reading frame}

\note{
   They used the \emph{Himar1\/} transposon which randomly inserts
   itself into a site {\hilit TA}.

   ``Stop codons'' are codes for where a gene stops. The point is that
   if this thing gets inserted into a gene, it will end up with a
   truncated product.
  }
\end{frame}





\begin{frame}[c]{Sequence of the gene MT598}

\figw{Figs/mt598.pdf}{1.0}

\note{
}

\end{frame}


\begin{frame}[c]{Random transposon mutagenesis}

\only<1|handout 0>{\figh{Figs/mut1.ps}{0.9}}
\only<2|handout 0>{\figh{Figs/mut2.ps}{0.9}}
\only<3|handout 0>{\figh{Figs/mut3.ps}{0.9}}
\only<4|handout 0>{\figh{Figs/mut4.ps}{0.9}}
\only<5|handout 0>{\figh{Figs/mut5.ps}{0.9}}
\only<6>{\figh{Figs/mut6.ps}{0.9}}

\note{
}

\end{frame}


\end{document}
