\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Building regression models}
\subtitle{Mapping multiple QTL}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\newcommand{\lod}{\text{LOD}}
\newcommand{\plod}{\text{pLOD}}
\newcommand{\M}{\text{M}}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
  In this lecture, we'll look at building regression models, and
  particularly at the case of trying to map multiple QTL. Our interest
  here is not in prediction but rather in identifying the important
  variables.
}
} }







\begin{frame}{Example}

\footnotesize
Sugiyama et al. Genomics 71:70-77, 2001

\bi
\item 250 male mice from the backcross (A $\times$ B) $\times$ B
\item Blood pressure after two weeks drinking water with 1\% NaCl
\ei

\figw{Figs/pheno.pdf}{1.0}

\end{frame}







\begin{frame}[c]{Genetic map}

\figw{Figs/geneticmap.pdf}{1.0}

\end{frame}





\begin{frame}[c]{Genotype data}

\figw{Figs/genodata.pdf}{1.0}

\end{frame}





\begin{frame}{Goals}

\bbi
\item Identify quantitative trait loci (QTL) \\[6pt]
   {\lolit   (and interactions among QTL)}
\item Interval estimates of QTL location
\item Estimated QTL effects
\ei

\end{frame}




\begin{frame}[c]{LOD curves}

\figw{Figs/alod.pdf}{1.0}

\end{frame}






\begin{frame}{Estimated effects}

\figw{Figs/meffects.pdf}{1.0}

\end{frame}






\begin{frame}{Modeling multiple QTL}

\bbi
\item Reduce residual variation $\longrightarrow$ increased power
\item Separate linked QTL
\item Identify interactions among QTL {\lolit (epistasis)}
\ei
\end{frame}






\begin{frame}[c]{Epistasis in BC}

\figw{Figs/epistasis_bc.pdf}{1.0}

\end{frame}






\begin{frame}[c]{Epistasis in F$_{\mathsf{2}}$}

\figw{Figs/epistasis_f2.pdf}{1.0}

\end{frame}





\begin{frame}[c]{2-dim, 2-QTL scan}

\hfill \begin{minipage}[t]{10in}
For all pairs of positions, fit the following models:

\vspace{10mm}

\hfill
\begin{minipage}{9in}
{\lolit
$\mathsf{\text{H}_f: y = \mu + \beta_1 q_1 + \beta_2 q_2 +
      \gamma q_1 q_2 + \epsilon}$

\vspace{5mm}

$\mathsf{\text{H}_a: y = \mu + \beta_1 q_1 + \beta_2 q_2 +
      + \epsilon}$

\vspace{5mm}

$\mathsf{\text{H}_1: y = \mu + \beta_1 q_1 + \epsilon}$

\vspace{5mm}

$\mathsf{\text{H}_0: y = \mu + \epsilon}$

}
\end{minipage}

\vspace{20mm}

log$_{\mathsf{10}}$ likelihoods:

\vspace{5mm}

\hfill
\begin{minipage}{9in}
{\lolit

$\mathsf{l_f(s,t)}$ \hspace{2cm}
$\mathsf{l_a(s,t)}$ \hspace{2cm}
$\mathsf{l_1(s)}$ \hspace{2cm}
$\mathsf{l_0}$ \hspace{2cm}
}
\end{minipage}


\end{minipage}


\end{frame}






\begin{frame}{2-dim, 2-QTL scan}


LOD scores:

\vspace{5mm}

\hspace{1cm}
\begin{minipage}{5in}
{\lolit
\begin{eqnarray*}
\mathsf{\lod_f(s,t)}& = &\mathsf{l_f(s,t) - l_0} \\[24pt]
\mathsf{\lod_a(s,t)}& = &\mathsf{l_a(s,t) - l_0} \\[24pt]
\mathsf{\lod_i(s,t)}& = &\mathsf{l_f(s,t) - l_a(s,t)} \\[24pt]
\mathsf{\lod_1(s)}&   = &\mathsf{l_1(s) - l_0} \\[24pt]
\end{eqnarray*}
}

\end{minipage}
\end{frame}







\begin{frame}[c]{Results: $\mathsf{\lod_i}$ and $\mathsf{\lod_f}$}

\figw{Figs/2dscan_allchr.pdf}{1.0}

\end{frame}







\begin{frame}[c]{Results: $\mathsf{\lod_i}$ and $\mathsf{\lod_f}$}

\figw{Figs/2dscan_selchr.pdf}{1.0}

\end{frame}







\begin{frame}{Summaries}

\hfill \begin{minipage}[t]{10in}
Consider each pair of chromosomes, {\lolit $\mathsf{(j, k)}$}, \\
and let {\lolit $\mathsf{c(s)}$} denote the chromosome for
  position {\lolit $\mathsf{s}$}.

\bigskip
\footnotesize
\begin{minipage}{6in}
{\lolit
\begin{eqnarray*}
\mathsf{\M_f(j,k)}& = &\mathsf{ \max_{c(s)=j, c(t)=k} \lod_f(s,t)} \\
\mathsf{\M_a(j,k)}& = &\mathsf{ \max_{c(s)=j, c(t)=k} \lod_a(s,t)} \\
\mathsf{\M_1(j,k)}& = &\mathsf{ \max_{c(s)=j \text{ or } k} \lod_1(s)} \\[12pt]
\mathsf{\M_i(j,k)}& = &\mathsf{ \M_f(j,k) - \M_a(j,k)}\\
\mathsf{\M_{fv1}(j,k)}& = &\mathsf{ \M_f(j,k) - \M_1(j,k)}\\
\mathsf{\M_{av1}(j,k)}& = &\mathsf{ \M_a(j,k) - \M_1(j,k)}
\end{eqnarray*}
}
\end{minipage}
\end{minipage}

\end{frame}







\begin{frame}[c]{Results: $\mathsf{\lod_i}$ and $\mathsf{\lod_{fv1}}$}

\figw{Figs/2dscan_selchr_fv1.pdf}{1.0}

\end{frame}







\begin{frame}{Thresholds}


\hfill \begin{minipage}[t]{9.5in}
A pair of chromosomes {\lolit $\mathsf{(j,k)}$} is considered interesting if:

\bigskip

{\lolit
\centerline{$ \mathsf{\M_f(j,k)} > \mathsf{ T_f }$ \hspace{5mm} {\hilit and }
\hspace{5mm}
{\hilit \{} $ \mathsf{\M_{fv1}(j,k)} > \mathsf{ T_{fv1} }$
{\hilit or }
$ \mathsf{\M_i(j,k)} > \mathsf{ T_i }$  {\hilit \}}}

\bigskip

\centerline{\hilit or }

\bigskip

\centerline{$ \mathsf{\M_a(j,k)} > \mathsf{ T_a }$ \hspace{5mm} {\hilit and }
\hspace{5mm}
$ \mathsf{\M_{av1}(j,k)} > \mathsf{ T_{av1} }$ }
}


\bigskip

where the thresholds {\lolit $\mathsf{(T_f, T_{fv1}, T_i, T_a,
    T_{av1})}$} are determined by a permutation test with a 2d scan

\end{minipage} \hspace{0.5in}


\end{frame}








\begin{frame}[fragile]{2d scan summary}



\begin{verbatim}
                   pos1f pos2f lod.full lod.fv1 lod.int
            c1:c4   71.3  30.0    14.36    6.78    0.27
            c6:c15  55.0  20.5     6.91    4.95    2.92
            c1:c1   39.3  78.3     5.10    1.58    0.09


                   pos1a pos2a  lod.add lod.av1
            c1:c4   68.3  30.0    14.09    6.50
            c6:c15  24.0  22.5     3.99    2.03
            c1:c1   48.3  79.3     5.02    1.50
\end{verbatim}



\end{frame}







\begin{frame}[c]{Estimated effects}

\figw{Figs/ieffects.pdf}{1.0}

\end{frame}








\begin{frame}[c]{Chr 1: $\mathsf{\lod_i}$ and $\mathsf{\lod_{av1}}$}

\figw{Figs/2dscan_chr1.pdf}{1.0}

\end{frame}







\begin{frame}{Hypothesis testing?}

\vspace{2cm}


\hfill \begin{minipage}[t]{10in}
\begin{itemize}
\itemsep20mm
\item In the past, QTL mapping has been regarded as a task of
  {\hilit hypothesis testing}.

\vspace{10mm}

\hspace{15mm} {\lolit Is this a QTL?}

\vspace{10mm}

Much of the focus has been on adjusting for test multiplicity.

\item It is better to view the problem as one of {\hilit model
  selection}.

\vspace{10mm}

\hspace{15mm} {\lolit What set of QTL are well supported?}

\hspace{15mm} {\lolit Is there evidence for QTL-QTL
  interactions?}

\vspace{10mm}

{\hilit Model} $\mathsf{=}$ a defined set of QTL and QTL-QTL interactions
\\
(and possibly covariates and QTL-covariate interactions).

\end{itemize}
\end{minipage}

\end{frame}









\begin{frame}[c]{Model selection}


  \begin{columns}
    \column{0.5\textwidth}

\bbi
\item Class of models
    \bi
\item Additive models
\item + pairwise interactions
\item + higher-order interactions
\item Regression trees
\ei

\item Model fit
 \bi
\item Maximum likelihood
\item Haley-Knott regression
\item extended Haley-Knott
\item Multiple imputation
\item MCMC
\ei

\ei

\column{0.5\textwidth}

\bbi
\item Model comparison
\bi
\item Estimated prediction error
\item AIC, BIC, penalized likelihood
\item Bayes
\item
\ei

\item Model search
\bi
\item Forward selection
\item Backward elimination
\item Stepwise selection
\item Randomized algorithms
\ei

\ei
\end{columns}

\end{frame}









\begin{frame}{Target}


\hfill \begin{minipage}{10in}
\begin{itemize}
\itemsep24pt
\item Selection of a model includes two types of errors:


\begin{quote} \begin{itemize}
\item Miss important terms (QTLs or interactions)
\item Include extraneous terms
\end{itemize} \end{quote}

\item Unlike in hypothesis testing, we can make {\hilit both errors} at
the same time.

\item {\color{myyellow} Identify as many correct terms as possible, while
{\hilit controlling the rate of inclusion of extraneous terms}.}

\end{itemize}
\end{minipage}

\end{frame}









\begin{frame}{What is special here?}


\bbi

\item Goal: identify the major players

\item A continuum of ordinal-valued covariates (the genetic loci)

\item Association among the covariates

\bi
\item Loci on different chromosomes are independent
\item Along chromosome, a very simple (and known) correlation
  structure
\ei

\ei

\end{frame}










\begin{frame}{Exploratory methods}


\bbi
\item Condition on a large-effect QTL


\bi
\item Reduce residual variation
\item Conditional LOD score:

\hspace{1in} $ \displaystyle{\mathsf{\lod(q_2 \ | \ q_1) = \text{log}_{10}
    \left\{\frac{\text{Pr}(\text{data} \ | \ q_1, q_2)}{
    \text{Pr}(\text{data} \ | \ q_1)}\right\} }}$
\ei


\item Piece together the putative QTL from the 1d and 2d scans

\bi
\item Omit loci that no longer look interesting (drop-one-at-a-time analysis)
\item Study potential interactions among the identified loci
\item Scan for additional loci (perhaps allowing interactions), conditional on these
\ei

\ei

\end{frame}




\begin{frame}[c]{Controlling for chr 4}

\figw{Figs/alod_c4.pdf}{1.0}

\end{frame}









\begin{frame}[fragile]{Drop-one-QTL table}


\vspace{40mm}


\begin{verbatim}
                                  df    LOD   %var
                1@68.3             1   6.30   11.0
                4@30.0             1  12.21   20.1
                6@61.0             2   7.93   13.6
                15@17.5            2   7.14   12.3
                6@61.0 : 15@17.5   1   5.68    9.9
\end{verbatim}




\end{frame}

\begin{frame}{Automation}


\hfill \begin{minipage}{10in}
\begin{itemize}
\itemsep24pt

\item Assistance to non-specialists

\item Understanding performance

\item Many phenotypes

\end{itemize}
\end{minipage}


\end{frame}






\begin{frame}{Additive QTL}


\hfill \begin{minipage}{10in}

Simple situation:

\lolit
\begin{itemize}
\item Dense markers
\item Complete genotype data
\item No epistasis

\end{itemize}

\vspace{2cm}

\centerline{
$\mathsf{y  = \mu + \sum \beta_j \, q_j + \epsilon}$ \hspace{1cm}
       {\hilit which $\mathsf{\beta_j \ne 0}$?}
}

\vspace{2cm}

{\color{myyellow}
$\mathsf{ \plod(\gamma) = \lod(\gamma) -
    {\hilit T} \, |\gamma| }$
}


\end{minipage}

\end{frame}






\begin{frame}{Additive QTL}


\hfill \begin{minipage}{10in}

Simple situation:

\lolit
\begin{itemize}
\item Dense markers
\item Complete genotype data
\item No epistasis

\end{itemize}

\vspace{2cm}

\centerline{
$\mathsf{y  = \mu + \sum \beta_j \, q_j + \epsilon}$ \hspace{1cm}
       {\hilit which $\mathsf{\beta_j \ne 0}$?}
}

\vspace{2cm}

{\color{myyellow}
$\mathsf{ \plod(\gamma) = \lod(\gamma) -
    {\hilit T} \, |\gamma| }$
}

\vspace{2cm}

\begin{minipage}[t]{1.4in}
\vspace*{0mm}

0 vs 1 QTL:
\end{minipage}
\begin{minipage}[t]{6in}
\vspace*{0mm}

\lolit
$\mathsf{\plod(\emptyset) = 0}$ \\[16pt]
$\mathsf{\plod(\{\lambda\}) =
    \lod(\lambda) - {\hilit T}}$
\end{minipage}


\end{minipage}



\end{frame}







\addtocounter{page}{-1}

\begin{frame}{Additive QTL}


\hfill \begin{minipage}{10in}

Simple situation:

\lolit
\begin{itemize}
\item Dense markers
\item Complete genotype data
\item No epistasis

\end{itemize}

\vspace{2cm}

\centerline{
$\mathsf{y  = \mu + \sum \beta_j \, q_j + \epsilon}$ \hspace{1cm}
       {\hilit which $\mathsf{\beta_j \ne 0}$?}
}

\vspace{2cm}

{\color{myyellow}
$\mathsf{ \plod(\gamma) = \lod(\gamma) -
    {\hilit T} \, |\gamma| }$
}

\vspace{2cm}

\begin{minipage}[t]{10in}
\vspace*{0mm}

For the mouse genome: \\[6pt]
\hspace*{0.5in} $\mathsf{\hilit T}$ = {\lolit
  2.69} (BC) or {\lolit 3.52} (F$_{\mathsf{2}}$)
\end{minipage}

\end{minipage}




\end{frame}








\begin{frame}{Experience}


\hfill \begin{minipage}{10in}

\begin{itemize}
\itemsep18pt
\item Controls rate of inclusion of extraneous terms
\item Forward selection over-selects
\item {\lolit Forward selection followed by backward elimination} works as well
  as {\lolit MCMC}

\item {\hilit Need to define performance criteria}
\item {\hilit Need large-scale simulations}
\end{itemize}

\vspace{6cm}


\hfill Broman \& Speed, JRSS B 64:641-656, 2002

\end{minipage}


\end{frame}







\begin{frame}{Epistasis}


\hfill \begin{minipage}{10in}

\centerline{
$\mathsf{y  = \mu + \sum \beta_j \, q_j + \sum \gamma_{jk} \, q_j \,
    q_k + \epsilon}$
}

\vspace{3cm}

{\color{myyellow}
$\mathsf{ \plod(\gamma) = \lod(\gamma) -
    {\hilit T_m} \, |\gamma|_m - {\hilit T_i} \, |\gamma|_i }$
}


\vspace{15mm}

\hspace{3cm} $\mathsf{\hilit T_m}$ = as chosen previously

\vspace{15mm}

\hspace{3cm} $\mathsf{\hilit T_i}$ = ?



\end{minipage}

\end{frame}






\begin{frame}{Idea 1}


\hfill \begin{minipage}{10in}

Imagine there are two additive QTL and consider a 2d, 2-QTL scan.

\vspace{1cm}

\hspace*{0.5in} $\mathsf{\hilit T_i}$ = 95th percentile of the
  distribution of \\[6pt]
\hspace*{1.3in} {\lolit $\mathsf{ \text{max} \, \lod_f(s,t) -
    \text{max} \, \lod_a(s,t)}$}


\end{minipage}



\end{frame}







\addtocounter{page}{-1}

\begin{frame}{Idea 1}


\hfill \begin{minipage}{10in}

Imagine there are two additive QTL and consider a 2d, 2-QTL scan.

\vspace{1cm}

\hspace*{0.5in} $\mathsf{\hilit T_i}$ = 95th percentile of the
  distribution of \\[6pt]
\hspace*{1.3in} {\lolit $\mathsf{ \text{max} \, \lod_f(s,t) -
    \text{max} \, \lod_a(s,t)}$}


\vspace{2cm}

For the mouse genome: \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T_m}$ = {\lolit
  2.69} (BC) or {\lolit 3.52} (F$_{\mathsf{2}}$) \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T^H_i}$ = {\lolit
  2.62} (BC) or {\lolit 4.28} (F$_{\mathsf{2}}$)


\end{minipage}



\end{frame}







\begin{frame}{Idea 2}


\hfill \begin{minipage}{10in}

Imagine there is one QTL and consider a 2d, 2-QTL scan.

\vspace{1cm}

\hspace*{0.5in} $\mathsf{\hilit T_m + T_i}$ = 95th percentile of the
  distribution of \\[6pt]
\hspace*{2.0in} {\lolit $\mathsf{ \text{max} \, \lod_f(s,t) -
    \text{max} \, \lod_1(s)}$}


\end{minipage}



\end{frame}







\addtocounter{page}{-1}

\begin{frame}{Idea 2}


\hfill \begin{minipage}{10in}

Imagine there is one QTL and consider a 2d, 2-QTL scan.

\vspace{1cm}

\hspace*{0.5in} $\mathsf{\hilit T_m + T_i}$ = 95th percentile of the
  distribution of \\[6pt]
\hspace*{2.0in} {\lolit $\mathsf{ \text{max} \, \lod_f(s,t) -
    \text{max} \, \lod_1(s)}$}


\vspace{2cm}

For the mouse genome: \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T_m}$ = {\lolit
  2.69} (BC) or {\lolit 3.52} (F$_{\mathsf{2}}$) \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T^H_i}$ = {\lolit
  2.62} (BC) or {\lolit 4.28} (F$_{\mathsf{2}}$) \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T^L_i}$ = {\lolit
  1.19} (BC) or {\lolit 2.69} (F$_{\mathsf{2}}$)


\end{minipage}

\end{frame}





\begin{frame}[c]{Models as graphs}

\figw{Figs/models.pdf}{1.0}

\end{frame}





\begin{frame}[c]{Results}

\figw{Figs/hyper_models1.pdf}{1.0}



\end{frame}

\addtocounter{page}{-1}

\begin{frame}{Results}

\vfill


\figw{Figs/hyper_models2.pdf}{1.0}



\end{frame}

\addtocounter{page}{-1}

\begin{frame}{Results}

\vfill


\figw{Figs/hyper_models3.pdf}{1.0}


\end{frame}

\begin{frame}{Profile LOD curves}

\vfill


\figw{Figs/lod_profile.pdf}{1.0}


\end{frame}









\begin{frame}[fragile]{Drop-one-QTL table}


\vspace{40mm}


\begin{verbatim}
                                  df    LOD   %var
                1@68.3             1   6.30   11.0
                4@30.0             1  12.21   20.1
                6@61.0             2   7.93   13.6
                15@17.5            2   7.14   12.3
                6@61.0 : 15@17.5   1   5.68    9.9
\end{verbatim}

\end{frame}


\begin{frame}[c]{Add an interaction?}

\figw{Figs/hyper_models4.pdf}{1.0}

\end{frame}









\addtocounter{page}{-1}

\begin{frame}[c]{Add an interaction?}

\figw{Figs/hyper_models5.pdf}{1.0}

\end{frame}









\addtocounter{page}{-1}

\begin{frame}[c]{Add an interaction?}

\figw{Figs/hyper_models6.pdf}{1.0}

\end{frame}









\addtocounter{page}{-1}

\begin{frame}[c]{Add an interaction?}

\figw{Figs/hyper_models7.pdf}{1.0}

\end{frame}









\addtocounter{page}{-1}

\begin{frame}[c]{Add an interaction?}

\figw{Figs/hyper_models8.pdf}{1.0}

\end{frame}










\begin{frame}[c]{Add another QTL?}

\figw{Figs/hyper_models9.pdf}{1.0}

\end{frame}











\addtocounter{page}{-1}

\begin{frame}[c]{Add another QTL?}

\figw{Figs/hyper_models10.pdf}{1.0}

\end{frame}










\addtocounter{page}{-1}

\begin{frame}[c]{Add another QTL?}

\figw{Figs/hyper_models11.pdf}{1.0}

\end{frame}






\begin{frame}[c]{Add a pair of QTL?}

\figw{Figs/hyper_models12.pdf}{1.0}

\end{frame}







\begin{frame}{Summary}


\hfill \begin{minipage}{9.5in}

\begin{itemize}
\itemsep24pt
\item QTL mapping is a model selection problem
\item The criterion for comparing models is most important
\item We're focusing on a penalized likelihood method, with penalties
  derived from permutation tests with 1d and 2d scans
\item Manichaikul et al., Genetics 181:1077--1086, 2009
\end{itemize}

\end{minipage} \hspace{0.5in}

\end{frame}

\end{document}
