\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Building regression models}
\subtitle{Mapping multiple QTL}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\newcommand{\lod}{\text{LOD}}
\newcommand{\plod}{\text{pLOD}}
\newcommand{\M}{\text{M}}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
  In this lecture, we'll look at building regression models, and
  particularly at the case of trying to map multiple QTL. Our interest
  here is not in prediction but rather in identifying the important
  variables.
}
} }




\begin{frame}[c]{LOD curves}

\figw{Figs/alod.pdf}{1.0}

\note{
  I've talked a lot about QTL mapping: seeking to identify the set of
  genetic loci in the genome that affect some quantitative trait, by
  doing an experimental cross and then performing ANOVA at each locus,
  one at a time.

  This basic analysis considers each predictor in isolation, though we
  expect there to be multiple loci that are influencing the trait.

  Today, I want to talk about efforts to go after multiple loci at
  once.
}
\end{frame}





\begin{frame}{Example}

\footnotesize
Sugiyama et al. Genomics 71:70-77, 2001

\bi
\item 250 male mice from the backcross (A $\times$ B) $\times$ B
\item Blood pressure after two weeks drinking water with 1\% NaCl
\ei

\figw{Figs/pheno.pdf}{1.0}

\note{
   As an example, I'll focus on this backcross of 250 mice, with the
   outcome being blood pressure. At each position, mice have one of
   two genotypes.
}
\end{frame}







\begin{frame}[c]{Genotype data}

\figw{Figs/genodata.pdf}{1.0}

\note{
  I like this example because the genotyping strategy makes the
  analysis quite difficult. The mice in this figure are sorted by
  blood pressure. There was a selective genotyping approach,
  in which only the 46 mice with highest blood pressure and the 46
  mice with lowest blood pressure were genotyped. But then they did
  an initial analysis and genotyped all the mice at regions of
  potential interest. And then they further genotyped apparent
  recombinant individuals in selected regions, to try to further hone
  in on QTL locations.
}
\end{frame}





\begin{frame}{Goals}

\bbi
\item Identify quantitative trait loci (QTL) \\[6pt]
   {\lolit   (and interactions among QTL)}
\item Interval estimates of QTL location
\item Estimated QTL effects
\ei

\note{
  Our goals are first to identify QTL and potentially interactions
  among QTL, and secondly to get interval estimates of their location
  and to further estimate their effects. But we're going to focus
  solely on the first of these.
}
\end{frame}




\begin{frame}[c]{LOD curves}

\figw{Figs/alod.pdf}{1.0}

\note{
  The genome scan results for this example shows a strong QTL on
  chromosome 4, and a double-humped QTL on chromosome 1. Next in line
  are QTL on chr 6 and 15, but both are quite far below the
  genome-wide threshold for statistical signifance.
}
\end{frame}






\begin{frame}{Estimated effects}

\figw{Figs/meffects.pdf}{1.0}

\note{
  Here are the estimated effects of these loci. For the QTL on
  chromosomes 1, 4, and 15, the A allele leads to lower blood
  pressure. This matches the parental strains' traits: the A strain
  has lower blood pressure than the B strain.

  On chr 6, though, the effect is in the opposite direction: the A
  allele causes an increase in blood pressure. This is called a
  {\hilit transgressive} QTL. The A strain has lower blood pressure,
  but it appears to include alleles that cause increased blood
  pressure, relative to the B allele.
}
\end{frame}






\begin{frame}{Modeling multiple QTL}

\bbi
\item Reduce residual variation $\longrightarrow$ increased power
\item Separate linked QTL
\item Identify interactions among QTL {\lolit (epistasis)}
\ei

\note{
  Why go after multiple QTL? First, by taking account of large-effect
  QTL, we can potentially increase our power to detect further, more
  modest effect loci.

  Second, we can better determine whether there are linked QTL, such
  as the loci on chromosome 1. Does that double-hump mean there are
  two QTL? The best way to determine that is to explicitly model two
  QTL and compare the best two-locus model to the best single-locus
  model.

  Third, there is the possibility of interactions between QTL. That
  the effect of one locus depends on the genotype at a second locus.
  We can't learn about such interactions without explicitly considering
  multiple QTL simultaneously.
}
\end{frame}






\begin{frame}[c]{Epistasis in BC}

\figw{Figs/epistasis_bc.pdf}{1.0}

\note{
   Genetics call interactions between two genetic loci ``{\hilit
     epistasis}.''

   Here, the dots are at the average phenotype for each of the 4
   possible two-locus genotypes in a backcross.  The difference
   between the dots on the blue curve is the effect of QTL 1 when
   QTL 2 is homozygous AA; the difference between the dots on the pink
   curve is the effect of QTL 1 when QTL 2 is heterozygous AB.

   In the left panel, the effect of QTL 1 is the same, irrespective of
   the genotype at QTL 2, and so the QTL are said to be ``{\hilit
     additive}.'' (Some people might say that they have {\hilit
     independent} effects, but I really don't like that terminology;
   much better to say {\hilit additive}.) Note that the effect of QTL
   2 is also the same, no matter the genotype at QTL 1.

   In the right panel, the effect of QTL of QTL 1 is larger when QTL 2
   is heterozygous than what QTL 2 is homozygous. Similarly for QTL 2:
   its effect depends on the genotype at QTL 1. So the QTL are said to
   be {\hilit epistatic}: they ``interact.''
}
\end{frame}






\begin{frame}[c]{Epistasis in F$_{\mathsf{2}}$}

\figw{Figs/epistasis_f2.pdf}{1.0}

\note{
   Here we illustrate epistasis in an intercross, where there are
   three possible genotypes at each QTL.

   In the left panel, the pattern of effect of QTL 1 is the same, no
   matter the genotype at QTL 2. So the QTL are additive.

   In the right panel, the pattern of effect of QTL 1 is different for
   different genotypes at QTL 2, and so the QTL are epistatic; they
   interact.

   Note the potential dependence on the scale of the phenotype. If you
   transformed the phenotype by taking logs or square-roots, the
   figure on the left would no longer have nice parallel curves.

   The figure on the right, though, cannot be repaired by transformation.
}
\end{frame}











\begin{frame}[c]{Estimated effects}

\figw{Figs/ieffects.pdf}{1.0}

\note{
   Back to our example data, the QTL on chr 1 and 4 look almost
   perfectly additive. The QTL on chr 6 and 15, however, show a strong
   interaction. Basically you see that if you're het at chr 6 and
   homozygous at chr 15 you have high blood pressure; otherwise you
   have low blood pressure.

   Note that the effect here is as large as the effect of the chr 4
   locus, but it's hard to see if you don't take the possibility of
   the interaction into account. It gets dampened by 1/2 if you
   consider the chr 6 and 15 loci separately.
}
\end{frame}







\begin{frame}{Model selection {\lolit (or variable selection)}}


  \bbi
\item Subset selection
\item L$_1$-penalized regression (the LASSO)
\item regression forests
\item Bayes
\item ...
  \ei

  \note{
    Ultimately, the question here is one of variable selection: which
    genetic loci seem to affect the phenotype, and how do they
    interact?

    There are lots of methods for model or variable selection, but I
    still prefer subset selection.

    And much of this work is focusing on minimizing prediction error.
    The idea is that all predictors have an effect, but there are so
    many of them and some have such small effects that it may be
    better to omit them, introducing a bit of bias but reducing the
    prediction variance.
  }
\end{frame}












\begin{frame}[c]{Model selection}


  \begin{columns}
    \column{0.5\textwidth}

\bi
\item Class of models
    \bi
\item Additive models
\item + pairwise interactions
\item + higher-order interactions
\item Regression trees
\ei

\bigskip

\item Model fit
 \bi
\item Maximum likelihood
\item Haley-Knott regression
\item extended Haley-Knott
\item Multiple imputation
\item MCMC
\ei

\ei

\column{0.5\textwidth}

\bi
\item Model comparison
\bi
\item Estimated prediction error
\item AIC, BIC, penalized likelihood
\item Bayes
\item[]
\ei

\bigskip

\item Model search
\bi
\item Forward selection
\item Backward elimination
\item Stepwise selection
\item Randomized algorithms
\ei

\ei
\end{columns}

\note{
}
\end{frame}









\begin{frame}{Target}


\bbi

\item Selection of a model includes two types of errors:
\bi
\item Miss important terms (QTLs or interactions)
\item Include extraneous terms
\ei

\item Unlike in hypothesis testing, we can make {\hilit both errors} at
the same time.

\item {\hilit Identify as many correct terms as possible, while \\
{\vhilit controlling the rate of inclusion of extraneous terms}.}
\ei

\note{
}
\end{frame}









\begin{frame}{What is special here?}


\bbi

\item Goal: identify the major players
  \bi
  \item {\vhilit not} prediction
  \ei

\item A continuum of ordinal-valued covariates (the genetic loci)

\item Association among the covariates

\bi
\item Loci on different chromosomes are independent
\item Along chromosome, a very simple (and known) correlation
  structure
\ei

\ei

\note{
}
\end{frame}










\begin{frame}{Exploratory methods}


\bbi
\item Condition on a large-effect QTL


\bi
\item Reduce residual variation
\item Conditional LOD score:

\hspace{3em} $ \displaystyle{\mathsf{\lod(q_2 \ | \ q_1) = \text{log}_{10}
    \left\{\frac{\text{Pr}(\text{data} \ | \ q_1, q_2)}{
    \text{Pr}(\text{data} \ | \ q_1)}\right\} }}$
\ei


\bigskip

\item Piece together the putative QTL from the 1d and 2d scans

\bi
\item Omit loci that no longer look interesting (drop-one-at-a-time analysis)
\item Study potential interactions among the identified loci
\item Scan for additional loci (perhaps allowing interactions), conditional on these
\ei

\ei

\note{
}
\end{frame}






\begin{frame}[c]{Controlling for chr 4}

\figw{Figs/alod_c4.pdf}{1.0}

\note{
}
\end{frame}









\begin{frame}[fragile,c]{Drop-one-QTL table}

\begin{verbatim}
                                  df    LOD   %var
                1@68.3             1   6.30   11.0
                4@30.0             1  12.21   20.1
                6@61.0             2   7.93   13.6
                15@17.5            2   7.14   12.3
                6@61.0 : 15@17.5   1   5.68    9.9
\end{verbatim}

\note{
}
\end{frame}






\begin{frame}{Automation}
\bbi
\item Assistance to non-specialists

\item Understanding performance

\item Many phenotypes
\ei

\note{
}
\end{frame}






\begin{frame}{Additive QTL}


\vspace{-2mm}

\bi
\item[] \hspace{-2em} Simple situation:

\bi
\item Dense markers
\item Complete genotype data
\item No epistasis
\ei \ei

\bigskip \bigskip

\centerline{
$\mathsf{y  = \mu + \sum \beta_j \, q_j + \epsilon}$ \hspace{1cm}
       {\hilit which $\mathsf{\beta_j \ne 0}$?}
}

\bigskip \bigskip

{\hilit
$\mathsf{ \plod(\gamma) = \lod(\gamma) -
    {\hilit T} \, |\gamma| }$
}


\bigskip

\only<2>{
0 vs 1 QTL:

{\lolit
\qquad $\mathsf{\plod(\emptyset) = 0}$ \\
\qquad $\mathsf{\plod(\{\lambda\}) =
    \lod(\lambda) - {\hilit T}}$
} }


\only<3|handout 0>{
For the mouse genome: \\
\qquad $\mathsf{\hilit T}$ = {\lolit
  2.69} (BC) or {\lolit 3.52} (F$_{\mathsf{2}}$)
}

\note{
}
\end{frame}








\begin{frame}{Experience}

\bbi
\item Controls rate of inclusion of extraneous terms
\item Forward selection over-selects
\item {\vhilit Forward selection followed by backward elimination} works as well
  as MCMC
\item {\hilit Need to define performance criteria}
\item {\hilit Need large-scale simulations}
\ei

\bigskip \bigskip \bigskip

\footnotesize
\hfill Broman \& Speed, JRSS B 64:641-656, 2002 \\
\hfill \href{https://doi.org/10.1111/1467-9868.00354}{\tt 10.1111/1467-9868.00354}

\note{
}
\end{frame}







\begin{frame}{Epistasis}

\bigskip \bigskip \bigskip

\centerline{
$\mathsf{y  = \mu + \sum \beta_j \, q_j + \sum \gamma_{jk} \, q_j \,
    q_k + \epsilon}$
}

\bigskip \bigskip \bigskip

{\hilit
$\mathsf{ \plod(\gamma) = \lod(\gamma) -
    {\hilit T_m} \, |\gamma|_m - {\hilit T_i} \, |\gamma|_i }$
}


\bigskip \bigskip \bigskip

\hspace{3em} $\mathsf{\hilit T_m}$ = as chosen previously

\bigskip

\hspace{3em} $\mathsf{\hilit T_i}$ = ?

\note{
}
\end{frame}






\begin{frame}{Idea 1}


\hfill \begin{minipage}{10in}

Imagine there are two additive QTL and consider a 2d, 2-QTL scan.

\vspace{1cm}

\hspace*{0.5in} $\mathsf{\hilit T_i}$ = 95th percentile of the
  distribution of \\[6pt]
\hspace*{1.3in} {\lolit $\mathsf{ \text{max} \, \lod_f(s,t) -
    \text{max} \, \lod_a(s,t)}$}



\only<2>{

\vspace{2cm}

For the mouse genome: \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T_m}$ = {\lolit
  2.69} (BC) or {\lolit 3.52} (F$_{\mathsf{2}}$) \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T^H_i}$ = {\lolit
  2.62} (BC) or {\lolit 4.28} (F$_{\mathsf{2}}$)
}

\end{minipage}

\note{
}
\end{frame}







\begin{frame}{Idea 2}


\hfill \begin{minipage}{10in}

Imagine there is one QTL and consider a 2d, 2-QTL scan.

\vspace{1cm}

\hspace*{0.5in} $\mathsf{\hilit T_m + T_i}$ = 95th percentile of the
  distribution of \\[6pt]
\hspace*{2.0in} {\lolit $\mathsf{ \text{max} \, \lod_f(s,t) -
    \text{max} \, \lod_1(s)}$}


\only<2>{

\vspace{2cm}

For the mouse genome: \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T_m}$ = {\lolit
  2.69} (BC) or {\lolit 3.52} (F$_{\mathsf{2}}$) \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T^H_i}$ = {\lolit
  2.62} (BC) or {\lolit 4.28} (F$_{\mathsf{2}}$) \\[12pt]
\hspace*{0.5in} $\mathsf{\hilit T^L_i}$ = {\lolit
  1.19} (BC) or {\lolit 2.69} (F$_{\mathsf{2}}$)

}

\end{minipage}

\note{
}
\end{frame}





\begin{frame}[c]{Models as graphs}

\figw{Figs/models.pdf}{1.0}

\note{
}
\end{frame}





\begin{frame}[c]{Results}

\only<1>{\figw{Figs/hyper_models1.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/hyper_models2.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/hyper_models3.pdf}{1.0}}

\note{
}
\end{frame}



\begin{frame}{Profile LOD curves}

\figw{Figs/lod_profile.pdf}{1.0}

\note{
}
\end{frame}




\begin{frame}[fragile,c]{Drop-one-QTL table}

\begin{verbatim}
                                  df    LOD   %var
                1@68.3             1   6.30   11.0
                4@30.0             1  12.21   20.1
                6@61.0             2   7.93   13.6
                15@17.5            2   7.14   12.3
                6@61.0 : 15@17.5   1   5.68    9.9
\end{verbatim}

\note{
}
\end{frame}




\begin{frame}[c]{Add an interaction?}

\only<1>{\figw{Figs/hyper_models4.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/hyper_models5.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/hyper_models6.pdf}{1.0}}
\only<4|handout 0>{\figw{Figs/hyper_models7.pdf}{1.0}}
\only<5|handout 0>{\figw{Figs/hyper_models8.pdf}{1.0}}

\note{
}
\end{frame}



\begin{frame}[c]{Add another QTL?}

\only<1>{\figw{Figs/hyper_models9.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/hyper_models10.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/hyper_models11.pdf}{1.0}}
\only<4|handout 0>{\figw{Figs/hyper_models12.pdf}{1.0}}

\note{
}
\end{frame}







\begin{frame}{Summary}


\bbi
\item QTL mapping is a model selection problem
\item The problem is finding the major players, not minimizing
  prediction error
\item The criterion for comparing models is most important
\item We're focusing on a penalized likelihood method, with penalties
  derived from permutation tests with 1d and 2d scans
\item Manichaikul et al., Genetics 181:1077--1086, 2009 \\
  \href{https://doi.org/10.1534/genetics.108.094565}{\tt doi:10.1534/genetics.108.094565}
\ei

\note{
}
\end{frame}



\end{document}
