\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Data diagnostics}
\subtitle{Cleaning genotype data in multi-parent populations}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
}
} }



\begin{frame}[c]{Diversity outbred mice}
\figw{../03_sims/Figs/hs.pdf}{1.0}

\note{
  I want to talk about data cleaning, and I'm going to do so focusing
  on a cleaning genotype data in Diversity Outbred mice. These mice
  are an advanced intercross derived from eight inbred founder
  strains.
}
\end{frame}


\begin{frame}{Diversity outbred mouse data}

  \bbi
  \item 500 DO mice
  \item GigaMUGA SNP arrays (114k SNPs)
  \item RNA-seq data on pancreatic islets
  \item Microbiome data (16S and shotgun sequencing)
  \item protein and lipid measurements by mass spec
  \item Collaboration with Alan Attie, Gary Churchill, Brian Yandell,
    Josh Coon, Federico Rey, and many others
    \ei

\note{
}

\end{frame}


\begin{frame}{Principles}

  \bbi
\item What might have gone wrong?
  \item How could it be revealed?
\only<2->{\item Also, just make a bunch of graphs.}
\only<3->{\item If you see something weird, try to figure it out.}
\ei

\note{
}

\end{frame}


\begin{frame}{Possible problems}

\bbi
\item Sample duplicates
\item Sample mix-ups
\item Bad samples
\item Bad markers
\item Genotyping errors in founders
  \ei

\note{
}

\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} What to look at first?}

\note{
}

\end{frame}


\begin{frame}[c]{Missing data per sample}
  \only<1|handout 0>{\figw{Figs/missing_data_per_sample.pdf}{1.0}}
  \only<2>{\figw{Figs/missing_data_per_sample_labeled.pdf}{1.0}}

  \note{

  }
\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} Swapped sex labels}

\note{
}

\end{frame}



\begin{frame}[c]{Ave SNP intensity on X and Y chr}
  \only<1|handout 0>{\figw{Figs/ave_snp_int_XY.pdf}{1.0}}
  \only<2|handout 0>{\figw{Figs/ave_snp_int_XY_select.pdf}{1.0}}
  \only<3>{\figw{Figs/ave_snp_int_XY_select_labeled.pdf}{1.0}}

\note{
}
\end{frame}

\begin{frame}[c]{Heterozygosity vs SNP intensity on X chr}
  \only<1|handout 0>{\figw{Figs/het_vs_Xint.pdf}{1.0}}
  \only<2>{\figw{Figs/het_vs_Xint_labeled.pdf}{1.0}}

\note{
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Sample duplicates}

\note{
}

\end{frame}



\begin{frame}[c]{Percent matching genotypes between pairs}
  \only<1|handout 0>{\figw{Figs/percent_matching_geno.pdf}{1.0}}
  \only<2>{\figw{Figs/percent_matching_geno_labeled.pdf}{1.0}}

\note{
}
\end{frame}





\begin{frame}[c]{}

\centerline{\Large \color{title} Sample mix-ups: RNA-seq data}

\note{
}

\end{frame}


\begin{frame}[c]{RNA-seq mix-ups}

\only<1|handout 0>{\figw{Figs/gve_scheme_1.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/gve_scheme_2.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/gve_scheme_3.pdf}{1.0}}
\only<4>{\figw{Figs/gve_scheme_4.pdf}{1.0}}

\note{
}

\end{frame}


\begin{frame}[c]{Distance matrix}

  \figh{Figs/gve_dist_matrix.png}{0.9}

\note{
}

\end{frame}

\begin{frame}[c]{Min vs self distance}
  \only<1|handout 0>{\figw{Figs/gve_best_vs_self.pdf}{1.0}}
  \only<2>{\figw{Figs/gve_best_vs_self_labeled.pdf}{1.0}}

\note{
}
\end{frame}

\begin{frame}[c]{RNA-seq mix-ups, details}
  \figw{Figs/gve_details.pdf}{1.0}

\note{
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Sample mix-ups: microbiome data}

\note{
}

\end{frame}


\begin{frame}[c]{Microbiome data}
  \figw{Figs/microbiome_scheme.pdf}{1.0}

\note{
}
\end{frame}

\begin{frame}{Sample mix-ups: Microbiome data}

\bbi
\item Impute genotypes at all SNPs in DNA samples

\item Map microbiome reads to mouse genome;\\ find reads overlapping a SNP

\item For each pair of samples (DNA + microbiome):

  \bi
  \itemsep10pt
  \item Focus on reads that overlap a SNP where \\
        that DNA sample is homozygous

  \item Distance = proportion of reads where SNP allele \\
    \hspace*{16mm} doesn't match DNA sample's genotype
  \ei
\ei

\note{
}
\end{frame}



\begin{frame}[c]{Microbiome DO361 vs DNA DO361}

  \centering
  \LARGE
\input{Tabs/dna361_mb361_table.tex}

\note{
}
\end{frame}


\begin{frame}[c]{Microbiome DO360 vs DNA DO360}

  \centering
  \LARGE
\input{Tabs/dna360_mb360_table.tex}

\note{
}
\end{frame}


\begin{frame}[c]{Microbiome DO360 vs DNA DO370}

  \centering
  \LARGE
\input{Tabs/dna370_mb360_table.tex}

\note{
}
\end{frame}


\begin{frame}[c]{Microbiome mix-ups: min vs self distance}

\only<1|handout 0>{\figw{Figs/microbiome_best_vs_self.pdf}{1.0}}
\only<2>{\figw{Figs/microbiome_best_vs_self_labeled.pdf}{1.0}}

\note{
}
\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} Sample quality}

\note{
}

\end{frame}


\begin{frame}[c]{Missing data per sample}
  \figw{Figs/missing_data_per_sample_labeled.pdf}{1.0}

\note{
  Again, the first thing to look at regarding sample quality is the
  amount of missing data per sample.
}
\end{frame}


\begin{frame}[c]{SNP array intensities}
  \figw{Figs/array_int.pdf}{1.0}

\note{
}
\end{frame}


\begin{frame}[c]{Allele frequencies by individual}
  \only<1|handout 0>{\figw{Figs/allele_freq_byind.pdf}{1.0}}
  \only<2>{\figw{Figs/allele_freq_byind_fixrange.pdf}{1.0}}

\note{
}
\end{frame}

\begin{frame}[c]{Genotype frequencies by individual}
  \only<1|handout 0>{\figw{Figs/geno_freq_byind.pdf}{1.0}}
  \only<2>{\figw{Figs/geno_freq_byind_labeled.pdf}{1.0}}

\note{
}
\end{frame}


\begin{frame}[c]{Genotype probabilities (one mouse on one chr)}
  \figh{Figs/geno_prob.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Genotype reconstruction (one mouse)}
  \figh{Figs/geno_reconstruct.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Percent missing vs. number of crossovers}
  \only<1|handout 0>{\figh{Figs/missing_v_nxo.png}{0.8}}
  \only<2|handout 0>{\figh{Figs/missing_v_nxo_labeled.png}{0.8}}
  \only<3|handout 0>{\figh{Figs/missing_v_nxo_subset.png}{0.8}}
  \only<4>{\figh{Figs/missing_v_nxo_subset_labeled.png}{0.8}}

\note{
}
\end{frame}


\begin{frame}[c]{Crossovers by generation}
  \figh{Figs/nxo_by_generation.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Percent genotyping errors}
  \only<1|handout 0>{\figh{Figs/percent_geno_errors.png}{0.8}}
  \only<2>{\figh{Figs/percent_geno_errors_subset.png}{0.8}}

\note{
}
\end{frame}




\begin{frame}[c]{}

\centerline{\Large \color{title} Marker quality}

\note{
}

\end{frame}



\begin{frame}[c]{Proportion missing data}
  \figh{Figs/missing_data_bymar.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Allele frequencies by marker}
  \figh{Figs/allele_freq_bymar.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Genotype frequencies by marker}
  \figh{Figs/geno_freq_bymar.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Genotyping error rates}
  \figh{Figs/geno_errors_bymar.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Nice markers}
  \figh{Figs/nice_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Crap markers}
  \figh{Figs/crap_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{More crap markers}
  \figh{Figs/more_crap_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{One bad blob}
  \figh{Figs/one_bad_blob_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Wrong genomic position}
  \figh{Figs/wrong_position_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Puzzling no calls}
  \figh{Figs/puzzling_nocalls_markers.png}{0.8}

\note{
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Founder genotyping errors}

\note{
}

\end{frame}


\begin{frame}[c]{One founder missing}
  \figh{Figs/one_founder_missing.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Another case}
  \figh{Figs/another_founder_missing.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Principles}

  \bbi
\item Think about what might have gone wrong, \\
  and how it might be revealed

 \item Order is important; cleaning one aspect may make it hard to see
   another

 \item Make lots of graphs

 \item If you see something weird, try to figure it out

 \item Don't trust; verify

   \ei

\note{
}

\end{frame}






\begin{frame}[c]{Summary}

  \bbi
\item Amount of missing data, as main indicator of problem
\item Sex swaps, sample duplicates, sample mix-ups
\item Identifying bad samples most important
\item Bad samples:
  \bi
\item Missing data
\item Heterozygosity
\item Number of crossovers
\item Number of genotyping errors
  \ei
\item Bad markers:
  \bi
\item Missing data
\item Number of genotyping errors
\item Observed vs inferred genotypes
  \ei
  \ei

\note{
}

\end{frame}




\begin{frame}[c]{Additional thoughts}

  \bbi
\item You often have to go back to the beginning and start over
\item Interactive graphs can speed things up
\item Do the work within a reproducible report
  \ei

\note{
}

\end{frame}



\end{document}
