\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Data diagnostics}
\subtitle{Cleaning genotype data in multi-parent populations}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
  In this lecture, we'll discuss data diagnostics: studying data to
  identify and hopefully correct problems. As usual for this class,
  we'll focus on a particular case study, of cleaning genotype data in
  diversity outbred mice.
}
} }



\begin{frame}[c]{Diversity outbred mice}
\figw{../03_sims/Figs/hs.pdf}{1.0}

\note{
  I want to talk about data cleaning, and I'm going to do so focusing
  on a cleaning genotype data in Diversity Outbred mice. These mice
  are an advanced intercross derived from eight inbred founder
  strains.
}
\end{frame}


\begin{frame}{Diversity outbred mouse data}

  \bbi
  \item 500 DO mice
  \item GigaMUGA SNP arrays (114k SNPs)
  \item RNA-seq data on pancreatic islets
  \item Microbiome data (16S and shotgun sequencing)
  \item protein and lipid measurements by mass spec
  \item Collaboration with Alan Attie, Gary Churchill, Brian Yandell,
    Josh Coon, Federico Rey, and many others
    \ei

\note{
   The data we're looking at concerns a set of 500 DO mice, as part of
   a collaboration with a bunch of investigators at UW-Madison plus
   Gary Churchill at the Jackson Lab. We have dense genotypes from
   SNP arrays, plus RNA-seq data on one tissue and microbiome data,
   mass spec data, and loads of clinical measurements.
}

\end{frame}


\begin{frame}{Principles}

  \bbi
\item What might have gone wrong?
  \item How could it be revealed?
\only<2->{\item Also, just make a bunch of graphs.}
\only<3->{\item If you see something weird, try to figure it out.}
\ei

\note{
  Is data cleaning something that is totally specific to a given
  context, with no general rules? Or are there commonalities between
  cleaning genotypes in mice and cleaning say electronic health record
  data?

  I declare that there are some general principles, and I think the
  first one is: imagine what could have gone wrong. And then next ask
  how it might be revealed in the data.

  Further, you should just make a ton of graphs, and if you see
  something odd, try to figure out what's going on.
}

\end{frame}


\begin{frame}{Possible problems}

\bbi
\item Sample duplicates
\item Sample mix-ups
\item Bad samples
\item Bad markers
\item Genotyping errors in founders
  \ei

\note{
  For these genotype data, these are the main problems: sample
  duplicates or mix-ups. Sample mixtures, even.

  Also samples being bad, or genetic markers being bad. Also
  genotyping errors in the founder strains.

  Could also be that markers are on the wrong chromosome.
}

\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} What to look at first?}

\note{
   A key question is what to look at first? What are the most
   fundamental problems, and how might we find them?
}

\end{frame}


\begin{frame}[c]{Missing data per sample}
  \only<1|handout 0>{\figw{Figs/missing_data_per_sample.pdf}{1.0}}
  \only<2>{\figw{Figs/missing_data_per_sample_labeled.pdf}{1.0}}

  \note{
    For genotype data, I think the first thing to look at is the
    amount of missing data. High proportions of missing data are
    usually an indication that a DNA sample was bad.

    The 500 mice were gathered in batches of 100. There were clear
    differences in the amount of missing data by batch. But mostly,
    there are a set of 7 samples with >10\% missing data, which should
    probably be tossed, as probably the remaining data is bad.
  }
\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} Swapped sex labels}

\note{
  The next thing that I always look at is whether the X chromosome
  genotypes for the mice matches what we would expect, given the
  reported sex.
}

\end{frame}



\begin{frame}[c]{Ave SNP intensity on X and Y chr}
  \only<1|handout 0>{\figw{Figs/ave_snp_int_XY.pdf}{1.0}}
  \only<2|handout 0>{\figw{Figs/ave_snp_int_XY_select.pdf}{1.0}}
  \only<3>{\figw{Figs/ave_snp_int_XY_select_labeled.pdf}{1.0}}

\note{
  I've used a lot of different techniques for verifying sex from X
  chromosome genotypes. But for these SNP array data, and probably
  also for sequence-based genotypes, the best thing seems to be to
  look at the intensity of alleles on the X and Y chromosomes. This is
  because the chromosome dosage effects are really strong. Females
  have two Xs and no Y chromosomes, while males have one of each.

  It turns out to be important to subset the X and Y chromosome
  markers to the ones that show a real sex difference. There seem to
  be a lot of markers that are annotated as being on the X or Y but
  don't actually have the expected dosage effects.

  Having done that, we see two clear blobs: males with high intensity
  on Y and low intensity on X, and females with the opposite. There
  are a couple of clear females that are really males, plus a couple
  of mice that look like XO females.
}
\end{frame}

\begin{frame}[c]{Heterozygosity vs SNP intensity on X chr}
  \only<1|handout 0>{\figw{Figs/het_vs_Xint.pdf}{1.0}}
  \only<2>{\figw{Figs/het_vs_Xint_labeled.pdf}{1.0}}

\note{
  Traditionally, I would focus on heterozygosity on the X chromosome.
  That females will have some heterozygous calls and males should have
  none. This plot shows heterozygosity on X, vs X chr intensity as
  before. The idea is that the vertical axis here is informative, but
  not so informative as the just the X and Y chr intensities
  themselves.

  It can take a while, and a lot of exploration, to come to the
  {\hilit best} diagnostics for a given problem.
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Sample duplicates}

\note{
  The next thing to look for is sample duplication. Are there pairs of
  samples with very similar genotype data?
}

\end{frame}



\begin{frame}[c]{Percent matching genotypes between pairs}
  \only<1|handout 0>{\figw{Figs/percent_matching_geno.pdf}{1.0}}
  \only<2>{\figw{Figs/percent_matching_geno_labeled.pdf}{1.0}}

\note{
  Here, I'll just look at all pairs of samples and calculation the
  proportion of matching genotypes. A histogram of that will
  many times show some pairs near 100\%.

  Here, the average is around 50\%. There are some pairs that are
  especially low; these all involve either DO306 or DO308, and are
  just due to those two samples being crappy. There are also some
  pairs that are at around 70\%. Those look to be siblings (which were
  supposed to be avoided here).
}
\end{frame}





\begin{frame}[c]{}

\centerline{\Large \color{title} Sample mix-ups: RNA-seq data}

\note{
  Next I look at sample mix-ups, starting with the RNA-seq data. Do
  the RNA-seq results seem to match the genotype data?
}

\end{frame}


\begin{frame}[c]{RNA-seq mix-ups}

\only<1|handout 0>{\figw{Figs/gve_scheme_1.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/gve_scheme_2.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/gve_scheme_3.pdf}{1.0}}
\only<4>{\figw{Figs/gve_scheme_4.pdf}{1.0}}

\note{
  My scheme for looking at this is very similar to the lecture last
  week. First, for each expression trait, look for a strong eQTL.

  In a twist relative to what I'd done last week, instead of trying to
  predict genotype from expresssion, let's instead predict expression
  from genotype. So then we have, for each mouse, a set of predicted
  expression values. Compare those to the observed expression values
  and calculate the correlation as a measure of similarity (or maybe
  the RMS difference as a measure of dissimilarity).
}

\end{frame}


\begin{frame}[c]{Distance matrix}

  \figh{Figs/gve_dist_matrix.png}{0.9}

\note{
  Turn those results into a distance matrix: for each RNA sample, how
  similar is it to each DNA sample? Here we have just under 400 RNA
  samples and 500 DNA samples. The leading diagonal should be saying
  the matching samples.
}

\end{frame}

\begin{frame}[c]{Min vs self distance}
  \only<1|handout 0>{\figw{Figs/gve_best_vs_self.pdf}{1.0}}
  \only<2>{\figw{Figs/gve_best_vs_self_labeled.pdf}{1.0}}

\note{
  We can plot the minimum in each row vs the value on the diagonal, we
  get this plot. Values along the diagonal are presumed correct.
  Values in the lower-right corner are wrong but there's some other
  sample that their close to. If you look at the sample IDs, you can
  probably see that there are 3 sample swaps down there.
}
\end{frame}

\begin{frame}[c]{RNA-seq mix-ups, details}
  \figw{Figs/gve_details.pdf}{1.0}

\note{
  Here we look at those six problem samples in more detail, by
  plotting their distance to each other sample. We see that these are
  clearly three pairs of sample swaps. We can't be sure whether the
  problem is in the DNA or in the RNA.
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Sample mix-ups: microbiome data}

\note{
  We also have a set of microbiome samples that could be used to
  investigate sample mix-ups.
}

\end{frame}


\begin{frame}[c]{Microbiome data}
  \figw{Figs/microbiome_scheme.pdf}{1.0}

\note{
  In this part of the study, they took mouse poop and extracted DNA
  and then did massive parallel sequencing. The goal was to
  characterize the populations of microorganisms in the gut of the
  mice, but the sequences also include many that were derived from the
  mouse house. The question is: do those mouse-derived sequences seem
  to match the SNP genotypes we have?
}
\end{frame}

\begin{frame}{Sample mix-ups: Microbiome data}

\bbi
\item Impute genotypes at all SNPs in DNA samples

\item Map microbiome reads to mouse genome;\\ find reads overlapping a SNP

\item For each pair of samples (DNA + microbiome):

  \bi
  \itemsep10pt
  \item Focus on reads that overlap a SNP where \\
        that DNA sample is homozygous

  \item Distance = proportion of reads where SNP allele \\
    \hspace*{16mm} doesn't match DNA sample's genotype
  \ei
\ei

\note{
  Our approach was first to impute the genome-wide SNP genotypes using
  the founder sequence data, and then to look at microbiome reads that
  overlap a SNP.  For each pair of samples (genotypes, microbiome), we
  focused on reads where the genotype data said homozygous, and then
  looked at whether the reads had the same or a different allele.
}
\end{frame}



\begin{frame}[c]{Microbiome DO361 vs DNA DO361}

  \centering
  \LARGE
\input{Tabs/dna361_mb361_table.tex}

\note{
  Here's a pair of samples both labeled DO361. The split up the SNPs
  into those where DO361 is homozygous for the major allele and those
  where it's homozygous for the minor allele. Then we look count the
  reads that overlap those SNPs, split according to whether they are
  showing the major or minor allele. Here we see nice correspondance.
  We can use the mismatches as an estimate of sequencing error, though
  it could also reflect errors in the SNP genotypes.
}
\end{frame}


\begin{frame}[c]{Microbiome DO360 vs DNA DO360}

  \centering
  \LARGE
\input{Tabs/dna360_mb360_table.tex}

\note{
  This sample, though, shows poor correspondance.
}
\end{frame}


\begin{frame}[c]{Microbiome DO360 vs DNA DO370}

  \centering
  \LARGE
\input{Tabs/dna370_mb360_table.tex}

\note{
  If we compare that microbiome sample to a different DNA sample,
  though, we can find one with good correspondance. And note that this
  is one of the pairs that showed up as a mix-up when we looked at the RNA-seq data.
}
\end{frame}


\begin{frame}[c]{Microbiome mix-ups: min vs self distance}

\only<1|handout 0>{\figw{Figs/microbiome_best_vs_self.pdf}{1.0}}
\only<2>{\figw{Figs/microbiome_best_vs_self_labeled.pdf}{1.0}}

\note{
  We can turn those frequencies of corresponding vs opposite alleles
  into a distance matrix, and then plot the value on the diagonal vs
  the minimum value in each row, as before. We find that most samples
  are fine, but there are also some clear mix-ups. And then there are
  a number of samples that are sort of in-between. Many of these turn
  out to look like mixtures. (And some of the good samples also really
  look like mixtures.)

  Of the mix-ups, there was just one in common between the microbiome
  data and the RNA-seq data. But while we have 500 genotyped mice,
  only 400 were assayed for RNA-seq, and 400 were assayed for
  microbiome, with only 300 in common. And the other mix-ups are in
  the portion that can't be triagulated.
}
\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} Sample quality}

\note{
   The next thing I look at is sample quality. Are there particular
   samples that are bad and should be omitted?
}

\end{frame}


\begin{frame}[c]{Missing data per sample}
  \figw{Figs/missing_data_per_sample_labeled.pdf}{1.0}

\note{
  Again, the first thing to look at regarding sample quality is the
  amount of missing data per sample.
}
\end{frame}


\begin{frame}[c]{SNP array intensities}
  \figw{Figs/array_int.pdf}{1.0}

\note{
   Also of interest is to look at the distribution of intensities on
   these SNP arrays. It's hard to look at 500 histograms, but actually
   you can get a pretty good picture of them by plotting density
   estimates. I've highlighted in pink the 7 samples that have very
   high rates of missing data. But in addition, you can sort of see
   two clusters of curves: a sort of typical curve (like all those in
   the first batch) plus curves that have somewhat reduced median and
   perhaps a heavy right tail.

   To more precisely identify the groups of curves, you might make a
   scatterplot of say the 1st percentile against the 99th percentile.
}
\end{frame}


\begin{frame}[c]{Allele frequencies by individual}
  \only<1|handout 0>{\figw{Figs/allele_freq_byind.pdf}{1.0}}
  \only<2>{\figw{Figs/allele_freq_byind_fixrange.pdf}{1.0}}

\note{

}
\end{frame}

\begin{frame}[c]{Genotype frequencies by individual}
  \only<1|handout 0>{\figw{Figs/geno_freq_byind.pdf}{1.0}}
  \only<2>{\figw{Figs/geno_freq_byind_labeled.pdf}{1.0}}

\note{
}
\end{frame}


\begin{frame}[c]{Genotype probabilities (one mouse on one chr)}
  \figh{Figs/geno_prob.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Genotype reconstruction (one mouse)}
  \figh{Figs/geno_reconstruct.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Percent missing vs. number of crossovers}
  \only<1|handout 0>{\figh{Figs/missing_v_nxo.png}{0.8}}
  \only<2|handout 0>{\figh{Figs/missing_v_nxo_labeled.png}{0.8}}
  \only<3|handout 0>{\figh{Figs/missing_v_nxo_subset.png}{0.8}}
  \only<4>{\figh{Figs/missing_v_nxo_subset_labeled.png}{0.8}}

\note{
}
\end{frame}


\begin{frame}[c]{Crossovers by generation}
  \figh{Figs/nxo_by_generation.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Percent genotyping errors}
  \only<1|handout 0>{\figh{Figs/percent_geno_errors.png}{0.8}}
  \only<2>{\figh{Figs/percent_geno_errors_subset.png}{0.8}}

\note{
}
\end{frame}




\begin{frame}[c]{}

\centerline{\Large \color{title} Marker quality}

\note{
}

\end{frame}



\begin{frame}[c]{Proportion missing data}
  \figh{Figs/missing_data_bymar.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Allele frequencies by marker}
  \figh{Figs/allele_freq_bymar.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Genotype frequencies by marker}
  \figh{Figs/geno_freq_bymar.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Genotyping error rates}
  \figh{Figs/geno_errors_bymar.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Nice markers}
  \figh{Figs/nice_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Crap markers}
  \figh{Figs/crap_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{More crap markers}
  \figh{Figs/more_crap_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{One bad blob}
  \figh{Figs/one_bad_blob_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Wrong genomic position}
  \figh{Figs/wrong_position_markers.png}{0.8}

\note{
}
\end{frame}

\begin{frame}[c]{Puzzling no calls}
  \figh{Figs/puzzling_nocalls_markers.png}{0.8}

\note{
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Founder genotyping errors}

\note{
}

\end{frame}


\begin{frame}[c]{One founder missing}
  \figh{Figs/one_founder_missing.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Another case}
  \figh{Figs/another_founder_missing.png}{0.8}

\note{
}
\end{frame}


\begin{frame}[c]{Principles}

  \bbi
\item Think about what might have gone wrong, \\
  and how it might be revealed

 \item Order is important; cleaning one aspect may make it hard to see
   another

 \item Make lots of graphs

 \item If you see something weird, try to figure it out

 \item Don't trust; verify

   \ei

\note{
}

\end{frame}






\begin{frame}[c]{Summary}

  \bbi
\item Amount of missing data, as main indicator of problem
\item Sex swaps, sample duplicates, sample mix-ups
\item Identifying bad samples most important
\item Bad samples:
  \bi
\item Missing data
\item Heterozygosity
\item Number of crossovers
\item Number of genotyping errors
  \ei
\item Bad markers:
  \bi
\item Missing data
\item Number of genotyping errors
\item Observed vs inferred genotypes
  \ei
  \ei

\note{
}

\end{frame}




\begin{frame}[c]{Additional thoughts}

  \bbi
\item You often have to go back to the beginning and start over
\item Interactive graphs can speed things up
\item Do the work within a reproducible report
  \ei

\note{
}

\end{frame}



\end{document}
