\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Data diagnostics}
\subtitle{Cleaning genotype data in multi-parent populations}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
}
} }



\begin{frame}[c]{Diversity outbred mice}
\figw{../03_sims/Figs/hs.pdf}{1.0}

\note{
  I want to talk about data cleaning, and I'm going to do so focusing
  on a cleaning genotype data in Diversity Outbred mice. These mice
  are an advanced intercross derived from eight inbred founder
  strains.
}
\end{frame}


\begin{frame}{Diversity outbred mouse data}

  \bbi
  \item 500 DO mice
  \item GigaMUGA SNP arrays (114k SNPs)
  \item RNA-seq data on pancreatic islets
  \item Microbiome data (16S and shotgun sequencing)
  \item protein and lipid measurements by mass spec
  \item Collaboration with Alan Attie, Gary Churchill, Brian Yandell,
    Josh Coon, Federico Rey, and many others
    \ei

\note{
}

\end{frame}


\begin{frame}{Principles}

  \bbi
\item What might have gone wrong?
  \item How could it be revealed?
\only<2->{\item Also, just make a bunch of graphs.}
\only<3->{\item If you see something weird, try to figure it out.}
\ei

\note{
}

\end{frame}


\begin{frame}{Possible problems}

\bbi
\item Sample duplicates
\item Sample mix-ups
\item Bad samples
\item Bad markers
\item Genotyping errors in founders
  \ei

\note{
}

\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} What to look at first?}

\note{
}

\end{frame}


\begin{frame}[c]{Missing data per sample}
  \only<1|handout 0>{\figw{Figs/missing_data_per_sample.pdf}{1.0}}
  \only<2>{\figw{Figs/missing_data_per_sample_labeled.pdf}{1.0}}

  \note{

  }
\end{frame}


\begin{frame}[c]{}

\centerline{\Large \color{title} Swapped sex labels}

\note{
}

\end{frame}



\begin{frame}[c]{Ave SNP intensity on X and Y chr}
  \only<1|handout 0>{\figw{Figs/ave_snp_int_XY.pdf}{1.0}}
  \only<2|handout 0>{\figw{Figs/ave_snp_int_XY_select.pdf}{1.0}}
  \only<3>{\figw{Figs/ave_snp_int_XY_select_labeled.pdf}{1.0}}

\note{
}
\end{frame}

\begin{frame}[c]{Heterozygosity vs SNP intensity on X chr}
  \only<1|handout 0>{\figw{Figs/het_vs_Xint.pdf}{1.0}}
  \only<2>{\figw{Figs/het_vs_Xint_labeled.pdf}{1.0}}

\note{
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Sample duplicates}

\note{
}

\end{frame}



\begin{frame}[c]{Percent matching genotypes between pairs}
  \only<1|handout 0>{\figw{Figs/percent_matching_geno.pdf}{1.0}}
  \only<2>{\figw{Figs/percent_matching_geno_labeled.pdf}{1.0}}

\note{
}
\end{frame}





\begin{frame}[c]{}

\centerline{\Large \color{title} Sample mix-ups: RNA-seq data}

\note{
}

\end{frame}


\begin{frame}[c]{RNA-seq mix-ups}

\only<1|handout 0>{\figw{Figs/gve_scheme_1.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/gve_scheme_2.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/gve_scheme_3.pdf}{1.0}}
\only<4>{\figw{Figs/gve_scheme_4.pdf}{1.0}}

\note{
}

\end{frame}


\begin{frame}[c]{Distance matrix}

  \figh{Figs/gve_dist_matrix.png}{0.9}

\note{
}

\end{frame}

\begin{frame}[c]{Min vs self distance}
  \only<1|handout 0>{\figw{Figs/gve_best_vs_self.pdf}{1.0}}
  \only<2>{\figw{Figs/gve_best_vs_self_labeled.pdf}{1.0}}

\note{
}
\end{frame}

\begin{frame}[c]{RNA-seq mix-ups, details}
  \figw{Figs/gve_details.pdf}{1.0}

\note{
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large \color{title} Sample mix-ups: microbiome data}

\note{
}

\end{frame}


\begin{frame}[c]{Microbiome data}
  \figw{Figs/microbiome_scheme.pdf}{1.0}

\note{
}
\end{frame}

% explanation of method

% tables (361 vs 361, 360 vs 360, 360 vs 370)

% plots unlabeled and labeled




\begin{frame}[c]{}

\centerline{\Large \color{title} Sample quality}

\note{
}

\end{frame}


% missing data per sample

% array intensities

% allele frequencies by individual

% SNP genotype frquencies by individual
% add labels

% heterozygosities, by individual

% genotype probabilities for one mouse

% genome reconstruction

% missing vs number of crossovers
% labeled
% blown up
% crossovers by generation

% percent genotyping errors by mouse
% blown up



\begin{frame}[c]{}

\centerline{\Large \color{title} Marker quality}

\note{
}

\end{frame}


% proportion missing data

% allele frequencies, by marker
% genotype frequencies, by marker
% heterozygosities, by marker

% genotyping error rates
% genotyping error rate vs percent missing
% on log scale


% intensity plots
% nice markers
% crap markers
% more crap markers
% one bad blob
% wrong genomic coordinates
% puzzling no calls




\begin{frame}[c]{}

\centerline{\Large \color{title} Founder genotyping errors}

\note{
}

\end{frame}

% one founder missing
% another case




\begin{frame}[c]{Summary}

\note{
}

\end{frame}



\begin{frame}[c]{Principles}

\note{
}

\end{frame}



\begin{frame}[c]{Additional thoughts}

\note{
}

\end{frame}



\end{document}
