\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}
\usepackage{eepic}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{The EM algorithm}
\subtitle{QTL mapping with a cure model}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
  In this lecture, I'll illustrate a couple of examples of the EM
  algorithm, via QTL mapping and with a phenotype that shows a common
  feature: of having a spike in its distribution. We'll look at a way
  to deal with that problem.
}

} }



\begin{frame}[c]{Intercross}
\figw{Figs/intercross.pdf}{1.0}

\note{
    QTL mapping is an effort to identify the genetic loci that
    contribute to variation in some quantitative trait, like blood
    pressure. Such loci are called quantitative trait loci (QTL).

    We start with two strains that differ in the trait of interest.
    That they show a consistent difference when raised in the same
    environment indicates that the difference is genetic. To try to
    identify genes contributing to the trait difference, we can
    perform a series of different crosses; the most common is the
    intercross.

    One gathers a number of intercross progeny, measures the trait,
    and then measures genotype at different positions along the
    chromosomes. We then look for positions where the genotype is
    associated with the phenotype.
}

\end{frame}





\begin{frame}[c]{QTL mapping}

\vspace{5mm}
\figw{Figs/lodcurve_insulin_with_effects.pdf}{0.96}

\note{
    So we will scan across the genome, at each position calculating
    some test statistic. Regions where the test statistic is large
    will show an association between genotype and phenotype.
}

\end{frame}




\begin{frame}[c]{Phenotype data}
\figw{Figs/pheno.pdf}{1.0}

\vspace{5mm}

{\lolit \footnotesize
Sugiyama et al.\ (2002) Physiol Genomics 10:5--12
}

\note{
    As an example, here are body weight and heart weight for 150 or so
    intercross mice.
}

\end{frame}


\begin{frame}[c]{Genotype data}
\figw{Figs/genodata.pdf}{1.0}

\note{
    In addition to that phenotype data, we'll have genotype data like
    this. For each mouse, we'll have data at a variety of positions
    across the genome, indicating homozygous blue, homozygous pink, or
    heterozygous (shown in green).

    One of the challenges is the missing data: the white pixels
    sprinkled into the data.
}

\end{frame}


\begin{frame}[c]{Genetic map}
\figw{Figs/geneticmap.pdf}{1.0}

\note{
   A genetic map will indicate the locations of the genetic markers
   along the chromosomes.

   Our genotype data is rather sparse. On a given chromosome, we get
   to see genotype only at 3-8 positions, with some pretty large gaps
   between these positions.
}

\end{frame}



\begin{frame}[c]{ANOVA at marker loci}

\begin{columns}

\column{0.5\textwidth}
\bi
\item Also known as {\hilit marker regression}.
\item Split mice into groups according to genotype at a marker.
\item Do a t-test / ANOVA.
\item Repeat for each marker.
\ei

\column{0.5\textwidth}

\figw{Figs/anova.pdf}{1.0}

\end{columns}

\note{
  The simplest analysis of this sort of data is to look at each
  marker, one at a time, split the mice into groups according to their
  genotype, and do analysis of variance.

  We look for markers where the different genotype groups show strong
  differences in average phenotype, as in the left panel for marker
  D15Mit184. We are not so interested in markers where there is little
  difference among the genotypes, as in the right panel for marker
  D2Mit92.
}

\end{frame}




\begin{frame}{ANOVA at marker loci}

\begin{columns}
\column{0.5\textwidth}

{\hilit Advantages}

\bi
\item Simple.
\item Easily incorporates covariates.
\item Easily extended to more complex models.
\item Doesn't require a genetic map.
\ei


\column{0.5\textwidth}

{\hilit Disadvantages}

\bi
\item Must exclude individuals with missing genotype data.
\item Imperfect information about QTL location.
\item Suffers in low density scans.
\item {\vhilit Only considers one QTL at a time.}
\ei

\end{columns}


\note{
  Advantages of this approach are that it's simple and doesn't require
  specialized software. You can easily incorporate covariates, but
  replacing the ANOVA with linear regression and including extra
  covariates. You can extend the analysis to a more complex model, for
  example Cox proportional hazards when you have censored survival
  times. You're just looking for association between the phenotype and
  a categorical covariate. (Then repeat for each individual marker.)
  You also don't need a genetic map for the markers; you're just
  looking at them individually.

  Disadvantages include that you need to omit individuals with missing
  genotype at a marker, and you get imperfect information about QTL
  location. The approach can suffer in low-density scans where the
  intervals between markers are large.

  The biggest disadvantage is that you're considering just one QTL at
  a time.
}

\end{frame}




\begin{frame}{Interval mapping}

{\hilit Lander \& Botstein (1989)}

\bbi
  \item Assume a {\hilit single} QTL model.
  \item Each position in the genome, one at a time, is posited as the
  putative QTL.
  \item Let $\mathsf{g = }$ 0/1/2 if the (unobserved) QTL genotype is
  AA/AB/BB. \\[12pt]
        Assume $\mathsf{y | g \sim N(\mu_g, \sigma)}$
  \item Given genotypes at linked markers, $\mathsf{y \sim}$ mixture of normal
  dist'ns with mixing proportions $\mathsf{\text{Pr}(g \ | \ \text{marker data})}$
\ei


\note{
  In 1989 Lander and Botstein came up with a method, now called
  ``interval mapping,'' to solve the first three disadvantages of
  doing ANOVA at markers.

  The idea is to use the same model as in ANOVA: that there is a
  single QTL that affects the trait and that the residual variation
  follows a normal distribution. But here, we allow the
  putative QTL to be at any arbitrary position. We'll consider each
  position in the genome, one at a time, as the QTL location, and scan
  along each chromosome.

  The challenge is that the genotype at the putative QTL will not be
  known. However, we can use surrounding markers to calculate the
  probability of the different possible QTL genotypes, given the
  available marker data.

  The phenotype, given QTL genotype, follows a normal distribution.
  Given the marker data (and not knowing QTL genotype), the phenotype
  follows a mixture of normal distributions with known mixing proportions.
}

\end{frame}

\begin{frame}[c]{Backcross}
\figw{Figs/backcross.pdf}{1.0}

\note{
  It's sometimes useful to think about a backcross, where you cross
  the two parents and then cross the F$_1$ back to one of the parents.
  All the offspring get a blue chromosome plus a recombinant
  chromosome, so at any one position they have one of two genotypes,
  homozygous blue or heterozygous. It can be easier to think about
  this, with just the two possible genotypes. For QTL mapping, we'd do
  the equivalent of a t-test.
}

\end{frame}


\begin{frame}{Genotype probabilities}


\only<1|handout 0>{\figw{Figs/genoprob1.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/genoprob2.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/genoprob3.pdf}{1.0}}
\only<4|handout 0>{\figw{Figs/genoprob4.pdf}{1.0}}
\only<5|handout 0>{\figw{Figs/genoprob5.pdf}{1.0}}
\only<6>{\figw{Figs/genoprob6.pdf}{1.0}}

\bigskip

{\small
Calculate {\hilit $\mathsf{\text{Pr}(g \ | \ \text{marker data})}$}, assuming

\sbi
\item No crossover interference
\item No genotyping errors
\ei

\bigskip

Or use the {\hilit hidden Markov model (HMM)} technology

\sbi
\item To allow for genotyping errors
\item To incorporate dominant markers
\item {\hilit (Still assume no crossover interference.)}
\ei
}


\note{
   A crucial ingredient to this interval mapping method is the
   probability of the different possible QTL genotypes, given the
   available marker data.

   Consider a particular position for a QTL, such as the location of
   the triangle, and consider the available genotypes along the
   recombinant chromosome in a backcross.

   A common assumption is that there is no crossover interference,
   meaning that the locations of exchanges are completely random,
   according to a Poisson process. So whether or not there is an
   exchange in one interval is indepedent of whether there is an
   exchange in any other interval.

   If we make that assumption, plus the assumption that the marker
   genotypes have no errors, then we just need to look at the two
   nearest typed markers, and we can do a little back-of-the envelope
   calculation to get the probability that the individual has an A
   allele at the QTL, given that they have B alleles at the two
   flanking positions.

   We can relax the ``no genotyping errors'' assumption with a hidden
   Markov model, but the no interference assumption is super
   convenient, though totally wrong.
}

\end{frame}





\begin{frame}{The normal mixtures}

\vspace{-5mm}

\begin{columns}
\column{0.5\textwidth}

\setlength{\unitlength}{0.18\textwidth}
\begin{center}
\begin{picture}(4.5,1)
\small
% lines
\Thicklines
\put(0.25,0.5){\line(1,0){4}}
\put(0.25,0.35){\line(0,1){0.3}}
\put(1.65,0.35){\line(0,1){0.3}}
\put(4.25,0.35){\line(0,1){0.3}}

% text
\put(0.25,0.1){\makebox(0,0){$\mathsf{M_1}$}}
\put(4.25,0.1){\makebox(0,0){$\mathsf{M_2}$}}
\put(1.65,0.1){\makebox(0,0){$\mathsf{Q}$}}
\put(0.95,0.8){\makebox(0,0){7 cM}}
\put(2.95,0.8){\makebox(0,0){13 cM}}
\end{picture} \end{center}
\vspace{5mm}

\sbi
\itemsep18pt
\item Two markers separated by 20 cM, with the QTL closer to the left marker.
\item The figure at right shows the distributions of the phenotype
conditional on the genotypes at the two markers.
\item The dashed curves correspond to the components of the mixtures.
\ei



\column{0.5\textwidth}

\vspace*{-5mm}

\figw{Figs/mixtures.pdf}{1.0}

\end{columns}

\note{
  Another way to think about these genotype probilities, is to think
  about the normal mixtures.
  Imagine a pair of markers in a backcross, with complete genotype
  data at the markers, and a QTL that's in the interval between them,
  a bit closer to the left marker.

  In one sense, there are two kinds of mice: those that are AB at the
  QTL, and those that are BB at the QTL. But considering the marker
  data, there are four kinds for mice. The mice that are AB at both
  markers will mostly also be AB at the QTL; their phenotypes will
  follow one normal distribution. The mice that are BB at both markers
  will largely be BB at the QTL, and their phenotypes will follow
  another normal distribution, shifted over a bit.

  The mice that are AB at the left marker and BB at the right marker
  will include a portion of mice that are AB at the QTL and follow the
  one normal distribution and another portion that are BB at the QTL
  and follow the other normal distribution; overall, their phenotypes
  will follow a blog-like distribution, the mixture of the two normal
  distributions.

  Similarly, mice that are BB at the left marker and AB at the right
  marker will have phenotypes following a mixture of normal
  distributions.

  Our goal will be to estimate the averages for the two distributions
  plus their common residual SD.
}


\end{frame}





\begin{frame}[c]{Interval mapping}


  \bbi
  \itemsep18pt

\item[] Let $\mathsf{p_{ij} = \text{Pr}(g_i = j | \text{marker data})}$

\item[] $\mathsf{y_i | g_i \sim N(\mu_{g_i},\sigma^2)}$

\item[] $\mathsf{\text{Pr}(y_i | \text{marker data},\mu_0,\mu_1,\sigma) =
\sum_j p_{ij} \, f(y_i; \mu_j,\sigma)}$

\qquad where $\mathsf{f(y; \mu,\sigma)= \exp[-(y-\mu)^2/(2\sigma^2)] / \sqrt{2 \pi
\sigma^2}}$


\item[] {\hilit Log likelihood}: \hspace{5mm}
$\mathsf{l(\mu_0,\mu_1,\sigma) = \sum_i \log \text{Pr}(y_i | \text{marker
data},\mu_0,\mu_1,\sigma)}$

\item[] Maximum likelihood estimates ({\hilit MLEs})
of $\mathsf{\mu_0}$, $\mathsf{\mu_1}$, $\mathsf{\sigma}$:

\qquad values for which $\mathsf{l(\mu_0,\mu_1,\sigma)}$ is maximized.
\ei


\note{

  So in interval mapping, we consider some single position for the one
  and only QTL, and then for each mouse we calculate the probabilities
  for each possible QTL genotype given the available marker data.

  The phenotypes are assumed to follow a normal distribution, given
  QTL genotype. Given the marker data, they follow a mixture of
  normals, with known mixing proportions.

  We can write down the log likelihood, which is a sum of log mixture
  probabilities, where those mixture probabilities are a sum over the
  individual components.

  The parameters to estimate are the phenotype averages for each QTL
  genotype, plus the residual SD. We seek the MLEs, which are the
  values for which the log likelihood is maximum.
}

\end{frame}




\begin{frame}{EM algorithm}


\vspace*{-8mm}
\bbi

\item[] {\hilit E step}:

\qquad Let \hspace{5mm} $w_{ij}^{(k)} = \text{Pr}(g_i = j |
y_i,\text{marker data},\hat{\mu}_0^{(k-1)},
\hat{\mu}_1^{(k-1)},\hat{\sigma}^{(k-1)})$ \\[12pt]

\hspace{29mm} $ = \frac{p_{ij} \, f(y_i; \hat{\mu}_j^{(k-1)},\hat{\sigma}^{(k-1)})}{ \sum_j p_{ij} \, f(y_i; \hat{\mu}_j^{(k-1)},\hat{\sigma}^{(k-1)})}$


\item[] {\hilit M step}:

\qquad Let \hspace{5mm} $\hat{\mu}_j^{(k)} = \sum_i y_i w_{ij}^{(k)} /
\sum_i w_{ij}^{(k)}$ \\[12pt]

\hspace{20mm} $\hat{\sigma}^{(k)} = \sqrt{ \sum_i \sum_j w_{ij}^{(k)} (y_i-\hat{\mu}_j^{(k)})^2/n}$


\item[] {\hilit The algorithm}: \\[6pt]

\qquad Start with $w_{ij}^{(1)} = p_{ij}$; iterate the E \& M steps until convergence.

\ei


\note{
   The EM algorithm is super for this problem, as if we knew the QTL
   genotype for each mouse, we could calculate the MLEs immediate as
   the phenotype averages for the mice grouped by QTL genotype, and
   then the residual SD.

   The EM algorithm will alternate between an E step where we
   calculate probabililties of each possible QTL genotype given not
   just the marker data but also the phenotype, and using the current
   estimates of the parameters, and then the M step where we
   re-estimate the parameters. Here the new estimates of the means
   will be weighted averages of the phenotypes, with the weights being
   the conditional probabilities from the E step. The residual SD will
   be estimated using a weighted sum of residuals.

   Rather that start with initial estimates, we actually start by
   jumping into the M step, using our initial QTL genotype
   probabilities as the weights.
}

\end{frame}



\begin{frame}[c]{\href{https://bit.ly/em_alg}{Interactive illustration}}

\figw{Figs/em_alg_illustration.png}{0.6}

\href{https://bit.ly/em_alg}{\lolit \footnotesize \tt bit.ly/em\_alg}


\note{
  I've prepared an interactive graph at {\tt bit.ly/em\_alg} that
  illustrates the EM algorithm in this case.

  On the y-axis are the phenotypes. And on the x-axis are the initial
  probabilities for the AB genotype given the marker data. When you
  click ``next'' it will step through the iterations of the EM
  algorithm. Using the available phenotype data, and with the
  estimated genotype-specific averages indicating that mice on the
  left (with BB genotype) have higher average phenotype than mice on
  the right (with AB genotype), the probabilities get titled toward
  the left. Mice with larger phenotypes get moved a bit to the left,
  and mice with smaller phenotypers get moved a bit to the right.

  After 4 iterations, there's no perceptible further movement.
}

\end{frame}



\begin{frame}[c]{LOD scores}

 The LOD score is a measure of the {\hilit strength of
evidence} for the presence of a QTL at a particular
location.

\bbi
\itemsep18pt

\item[] $\text{LOD}(\lambda) = \log_{10}$ likelihood ratio comparing the hypothesis of a \\

\hspace{25mm} QTL at position $\lambda$ versus that of no QTL \\[8pt]

\hspace{14mm} $= \log_{10} \left\{ \frac{\text{Pr}(y | \text{QTL at $\lambda$}, \hat{\mu}_{0\lambda},
\hat{\mu}_{1\lambda}, \hat{\sigma}_\lambda)}{\text{Pr}(y | \text{no QTL}, \hat{\mu},
\hat{\sigma})} \right\}$

 \item[] $\hat{\mu}_{0\lambda}, \hat{\mu}_{1\lambda}, \hat{\sigma}_\lambda$ are the MLEs,
assuming a single QTL at position $\lambda$.


 \item[] {\hilit No QTL model:}

\bi
  \item[] {\color{foreground} The phenotypes are independent and identically
distributed (iid) $\text{N}(\mu, \sigma^2)$}.
\ei

\ei


\note{
   For historical reasons, we measure evidence for QTL using LOD
   scores: the log$_{10}$ likelihood ratio comparing the hypothesis of
   a single QTL at the given location to the null hypothesis of no QTL.
}

\end{frame}





\begin{frame}[c]{\href{https://bit.ly/D3lod}{Interactive plot}}

\figw{Figs/interactive_lod_curve.png}{0.65}

\href{https://bit.ly/D3lod}{\lolit \footnotesize \tt bit.ly/D3lod}


\note{
  Plotting these LOD scores across the genome was perhaps the biggest
  innovation of Lander and Botstein (1989).

  I've created an interactive graph to illustrate these LOD curves. In
  this case, we are also splitting the mice by sex. Click on a
  chromosome at the top and you'll see a blown-up view on the bottom
  left. Click on different positions along the chromosome and you can
  see the phenotype:genotype relationship.

  It's important to look at the phenotype:genotype associations and
  not just the summary LOD curves. While for the locus on chromosome
  2, the RR genotype is associated with higher insulin level (which
  corresponds to what is seen in the B and R founder strains), if you
  click on chromosome 9 you'll see that there the effect is in the
  opposite direction. And if you look at chr 7, you'll see that the
  effect is much larger in males than in females.
}

\end{frame}





\begin{frame}{Interval mapping}

\begin{columns}

\column{0.5\textwidth}

{\hilit Advantages}

\bi
\item Takes proper account of missing data.
\item Allows examination of positions between markers.
\item Gives improved estimates of QTL effects.
\item Provides pretty graphs.
\ei


\column{0.5\textwidth}

{\hilit Disadvantages}

\bi
\item Increased computation time.
\item Requires specialized software.
\item Difficult to generalize.
\item {\vhilit Only considers one QTL at a time.}
\ei


\end{columns}


\note{
  This interval mapping approach solves many of the weakness of just
  doing ANOVA at the markers: it takes account of missing data, it
  allows you to inspect positions between markers, at it gives
  improved estimates of QTL effects because you're looking directly at
  the QTL position. The biggest innovation, though, is graphing the
  test statistic across the genome.

  Disadvantages include that it requires increased computation time,
  it requires special software, and it can be difficult to generalize.
  These aren't so important anymore.

  The main disadvantage is that you're still just thinking about a
  single QTL. If you have reasonably dense markers and reasonably
  complete genotype data, this is really no different than just doing
  ANOVA with each individual marker.
}

\end{frame}




\begin{frame}[c]{Survival after Listeria infection}
\figw{Figs/listeria_hist.pdf}{1.0}

\note{
}

\end{frame}


\begin{frame}{Normal assumption in ANOVA}

  \bbi
\item ANOVA is remarkably robust
\item Transformation
\item Rank-based methods
\item Specially-tailored models (e.g. GLM)
  \ei


\note{
}

\end{frame}




\begin{frame}[c]{}

\centerline{\Large \color{title} Censoring?}


\note{
}

\end{frame}


\begin{frame}{Measurements with a spike at 0}

\bbi
\item Mass of gallstones

\item Gene expression, when a gene might be turned off

\item Microbiome data, when a microbe might be absent

\item Area of garage
\ei


\note{
}

\end{frame}





\begin{frame}{Two-part (``cure'') model}

  \bbi
  \itemsep24pt

  \item Let $z_i$ = 1 if mouse $i$ survived the infection \\[12pt]

  {\color{background} Let} $y_i$ = survival time

\item Assume $\text{Pr}(z_i|g) = \pi_g$ \\[12pt]

  {\color{background} Assume} $y_i | z_i=$ 0, $g \sim$ Normal($\mu_g$,
  $\sigma$) \\[12pt]

  {\color{background} Assume} $\{(y_i, z_i, g)\}$ mutually independent

  \ei


\note{
}

\end{frame}



\begin{frame}{EM algorithm}

  \begin{columns}

    \column{0.5\textwidth}

    {\hilit E step}

    \bigskip

    \figw{Figs/twopart_estep.png}{1.0}


    \column{0.5\textwidth}

    {\hilit M step}

    \bigskip

    \figw{Figs/twopart_mstep.png}{1.0}


  \end{columns}


\note{
}

\end{frame}



\begin{frame}{Tests}

  \bbi
\item $\pi_{AA} = \pi_{AB} = \pi_{BB}$
\item $\mu_{AA} = \mu_{AB} = \mu_{BB}$
\item $\pi_{AA} = \pi_{AB} = \pi_{BB}$ and $\mu_{AA} = \mu_{AB} = \mu_{BB}$
  \ei



\note{
}

\end{frame}


\begin{frame}[c]{LOD curves}
\figw{Figs/listeria_lod.pdf}{1.0}

\note{
}

\end{frame}


\begin{frame}[c]{QTL effects}
\figw{Figs/listeria_effects.pdf}{1.0}

\note{
}

\end{frame}




\begin{frame}{Lesson\only<2>{s}}

\bbi
\item Don't just cram your data into the standard approach.

\only<2->{
\item Cramming your data into the standard approach might work fine. }

\ei


\note{
}

\end{frame}

\begin{frame}{Standard approach}
\only<1|handout 0>{\figw{Figs/listeria_stdlod.pdf}{1.0}}
\only<2>{\figw{Figs/listeria_stdlodB.pdf}{1.0}}

\note{
}

\end{frame}




\begin{frame}{References}
\vspace{-7mm}

  \bbi

\item Lander ES, Botstein D (1989) Mapping Mendelian factors
  underlying quantitative traits using RFLP linkage maps. Genetics
  121:185-199 \\
  \href{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1203601}{\footnotesize
    PMCID: PMC1203601}

\item Broman KW (2001) Review of statistical methods for QTL mapping
  in experimental crosses. Lab Animal 30(7):44-52 \\
  \href{https://www.ncbi.nlm.nih.gov/pubmed/11469113}{\footnotesize
    PMID: 11469113}

\item Boyartchuk VL, et al. (2001) Multigenic control of Listeria monocytogenes
  susceptibility in mice. Nat Genet 27:259-260 \\
  \href{https://doi.org/10.1038/85812}{\footnotesize doi:10.1038/85812}

\item Broman KW (2003) Mapping quantitative trait loci in the case
  of a spike in the phenotype distribution. Genetics 163:1169-1175 \\
  \href{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1462498}{\footnotesize
    PMCID: PMC1462498}

\ei



\note{
  Here are some papers related to the work I've discussed today.

  Lander and Botstein (1989) introduced this idea of interval mapping
  of QTL, and the use of the EM algorithm to deal with missing
  genotype data in this context.

  The second is a gentler review of QTL mapping.

  The last two papers concern this ``spike in the phenotype
  distribution'' situation.
}

\end{frame}

\end{document}
