\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Permutation tests}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\begin{document}

{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
  In this lecture, we'll look at permutation tests. When they are
  appropriate, I prefer them.
}
} }



\begin{frame}[c]{Randomized experiment}

\figh{Figs/experiment.pdf}{0.9}

\note{
   Consider a randomized experiment with two treatment groups, treated
   (T) and control (C).

   How to tell whether the treatment has an effect?
}
\end{frame}



\begin{frame}[c]{Experimental results}

\figh{Figs/experiment_results.pdf}{0.9}

\note{
  The standard t-test gives a p-value of 0.01.

  What assumptions are made here?

  I find the permutation test to be more natural, and it only relies
  on the assumption of random assignment of treatment groups.

  In a permutation test, you compare the observed test statistic to
  the distribution of values you get when the treatment group
  assignments are shuffled/randomized/permuted.
}
\end{frame}




\begin{frame}[c]{Permuted results}

\only<1>{\figh{Figs/perm_results1.pdf}{0.9}}
\only<2|handout 0>{\figh{Figs/perm_results2.pdf}{0.9}}
\only<3|handout 0>{\figh{Figs/perm_results3.pdf}{0.9}}
\only<4|handout 0>{\figh{Figs/perm_results4.pdf}{0.9}}

\note{
  Here are results when you permute the treatment assignments. Do the
  observed results show a sufficiently strong effect that we can be
  confident that it's real?
}
\end{frame}




\begin{frame}[c]{10,000 permutations}

\only<1|handout 0>{\figh{Figs/perm_hist.pdf}{0.9}}
\only<2|handout 0>{\figh{Figs/perm_hist_with_t.pdf}{0.9}}
\only<3>{\figh{Figs/perm_hist_with_arrow.pdf}{0.9}}

\note{
  Here are the t statistics from 10,000 permutations of our data.

  We've superposed a t distribution with 22 degrees; note how closely
  it matches.

  RA Fisher noted the close correspondance between the theoretical
  t-distribution and the permutation distribution, and used this to
  justify use of probabilities from the t-distribution. The
  permutation results were what he wanted, but they were too difficult
  to obtain at the time, and the t-distribution provided a good
  approximation.
}
\end{frame}



\begin{frame}[c]{Assumptions for the permutation test}

\centering

  {\large The observations are {\vhilit exchangeable} \\[8pt]
    under the null hypothesis.}

\note{
  The only assumption for the permutation test is that the
  observations are exchangeable. Basically this means that the labels
  don't matter. It's a weaker assumption than that they are
  independent and identically distributed.

  For a randomized experiment, this is true by design.

  Basically you want the data to be as if they were assigned to
  treatment groups at random.
}
\end{frame}





\begin{frame}{What test statistic?}

  \bbi
  \item Anything will be {\hilit valid}
  \item Focus on {\vhilit power}
  \item Robustness can still be important
    \bi
  \item[] For example, resistance to outliers
    \ei
  \ei

\note{
  Here, we used the t statistic. But you can use {\hilit any}
  statistic you want with the permutation test.

  Much of the time, we choose statistics based on their null
  distribution being something we can approximate. But we don't care
  about that, since we can simulate to get an approximation of the
  permutation distribution.

  The focus is on {\hilit power}: what statistic will best show the
  expected effect? But note that we may still need to worry about
  robustness, as things like outliers can distort the permutation
  distribution and so weaken our ability to see real effects.
}
\end{frame}




\begin{frame}{How many permutations?}

  \bbi
  \item Typically n = 1,000 or 10,000
  \item Focus on getting a good estimate of the p-value
  \item X = number of permutations $\ge$ observed value \\[8pt]
    \hspace{3mm} $\sim$ binomial(n, p) where p = true p-value
  \item With small datasets, may be able to do an {\hilit exhaustive enumeration}.
    \ei

\note{
    How many permutation replicates to do? I view it as an effort to
    estimate the p-value, or to estimate the significance threshold
    with $\alpha=0.05$ or so.

    Typically I'll do 1,000 or 10,000. In some cases (for example,
    when I'm trying to control some false discovery rate), I may need
    to do many more.
}
\end{frame}



\begin{frame}[c]{}
\figh{Figs/churchill_doerge.png}{0.95}
\note{
  This paper introduced the idea of using a permutation test to assess
  signifiance in QTL mapping. The key issue is in controlling for the
  scan across the genome.
}
\end{frame}



\begin{frame}[c]{QTL data}

\figh{Figs/geno_and_pheno.pdf}{0.9}

\note{
  Here are some example QTL data: genotypes across the genome for a
  set of individuals, plus a quantitative phenotype.

  Note the correlations in genotypes within each chromosome.
}
\end{frame}


\begin{frame}[c]{QTL genome scan}

\only<1|handout 0>{\figh{Figs/qtl_scan.pdf}{0.9}}
\only<2|handout 0>{\figh{Figs/perm_scan_1.pdf}{0.9}}
\only<3|handout 0>{\figh{Figs/perm_scan_2.pdf}{0.9}}
\only<4|handout 0>{\figh{Figs/perm_scan_3.pdf}{0.9}}
\only<5>{\figh{Figs/perm_scan_4.pdf}{0.9}}

\note{
  Here are the QTL mapping results. The gray curves are the results
  for four successive permutations of the rows of the genotypes
  relative to the phenotypes.
}
\end{frame}


\begin{frame}[c]{Permutation results}

\figh{Figs/qtl_perm_results.pdf}{0.9}

\note{
  For each permutation, we take the genome-wide maximum test
  statistic.

  I find it important to always look at the distribution of
  these maxima. It should always look like this, sort of a chi-square
  distribution. If the signifiance threshold is unusually high, it
  could be that there's something weird in the data that's causing
  problems.

  We can pick off our value in this distribution and calculate a
  p-value that adjusts for the scan across the genome. Or we can use
  the 95th or 99th or 90th percentile as a significance threshold.
}
\end{frame}


\begin{frame}{Multiple testing}

  \bbi
  \item {\vhilit Many} examples
    \bi
    \item gene expression or proteomic studies
    \item genome-wide association studies
    \item 1000s of predictors in an epi study
    \ei
  \item Most stringent approach: control family-wise error rate (FWER)
  \item A Bonferroni adjustment can be too conservative
  \item Take max statistic in each permutation replicate
    \ei

\note{
   Adjusting for the genome scan in QTL mapping is one instance of the
   ``multiple testing'' problem. We use the most stringent approach,
   of controlling the ``family-wise error rate'' where ``family''
   refers to the set of hypotheses (the genomic positions).

   A Bonferroni adjustment is common (multiply the p-values by the
   number of tests) but it can be much too conservative when there are
   many correlated tests.

   Adjusting for the number of tests can be easy in the context of a
   permutation test, by taking the maximum statistic.
}
\end{frame}


\begin{frame}{If test statistic varies}

  \bbi
  \item taking max(X$_{\text{j}}$) assumes that the X$_{\text{j}}$ have a common null
    distribution

  \item if not, you'd want to normalize so they do

  \item One approach: use the permutation results to do so

  \bi \item for each column of permutation results, turn values into ranks
  \item then find the maximum rank in each row
  \item find where the observed statistics rank within each column
    \item This gives adjusted p-values that account for the search
      \ei
      \ei

\note{
  The max statistic assumes that the individual test statistics are
  similar in distribution. If that's not true, they may need to be
  normalized so that they are.

  One way to do this is to use the permutation results to force each
  statistic to have the same marginal distribution. The simplest way
  to do this is to turn things into ranks and then take the maximum
  rank for each permutation replicate.

  (Note there will be a need to increase the number of permutation replicates.)
}
\end{frame}




\begin{frame}{Abuse of p-values}

\vspace{-12pt}

  \bbi
\item Focusing on strict, arbitrary thresholds like 0.05
\item Not looking at the confidence interval for the effect
\item Ignoring multiple comparisons
\item Turning science into true/false questions
  \ei

  \only<2->{

    \centering
    \large


    \vspace{36pt}

    But I still like p-values.

  }

  \only<3->{

    \vspace{18pt}

    It's useful to ask, ``{\hilit Could this just be noise?}''

  }

\note{
  P-values are much maligned. And permutation tests are really all
  about p-values.

  But I still like them. I find it useful to ask, ``Could this all
  just be noise?'' If the answer is ``Yes,'' maybe you shouldn't
  bother doing much further.
}

\end{frame}


\begin{frame}[c]{Randomized block design}


\end{frame}



\begin{frame}[c]{Selective genotyping}

\figh{Figs/sel_geno.png}{0.85}

\note{
  A common strategy in QTL mapping to save on the cost of genotyping
  is to just genotype the top and bottom portion of subjects, by
  phenotype.
}
\end{frame}




\begin{frame}[c]{}
\figh{Figs/manichaikul_sig_thr.png}{0.95}
\note{
  A friend asked about what to do for significance thresholds in the
  case of selective genotyping. This is a very short paper
  demonstrating the use of a stratified permutation test.
}
\end{frame}



\begin{frame}{Summary}

  \bbi
\item Permutation tests, when appropriate, are the most natural
  of significance test.
\item Permutation tests can make it easy to control for multiple
  testing.
\item Stratified permutation tests accommodate a common
  non-exchangeable situtation.
\item Many are quite negative about p-values, but I still like them.
  \ei

\note{
  Always good to have a summary.
}
\end{frame}

\end{document}
