\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Computer simulations}
\subtitle{The genomes of recombinant inbred lines}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics \\ Univ.\ Wisconsin{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \note{}

} }



\begin{frame}[c]{}

\vspace*{-1mm} \hspace*{-2mm}
\figw{Figs/inbredmice.jpg}{1.2}

\end{frame}




\begin{frame}{}

\vspace*{18mm}

\centerline{
\begin{minipage}[t]{50mm}
\includegraphics[height=50mm]{Figs/da-vinci-man.jpg}
\end{minipage}
\hspace{15mm}
\begin{minipage}[t]{50mm}
\includegraphics[height=50mm]{Figs/vitruvian_mouse.jpg}
\hspace{5mm}
\href{http://daviddeen.com}{\scriptsize \lolit \tt daviddeen.com}
\end{minipage}
}
\end{frame}





\begin{frame}[c]{Intercross}
\figw{Figs/intercross.pdf}{1.0}
\end{frame}





\begin{frame}[c]{QTL mapping}

\vspace{5mm}
\figw{Figs/lodcurve_insulin_with_effects.pdf}{0.96}
\end{frame}


\begin{frame}[c]{Congenic line/NIL}

\figw{Figs/congenic.pdf}{1.0}

\end{frame}




\begin{frame}[c]{Advanced intercross lines}

  \figw{Figs/ail.pdf}{1.0}

\end{frame}


\begin{frame}[c]{Recombinant inbred lines}

  \only<1>{\figw{Figs/rilines.pdf}{1.0}}
  \only<2>{\figw{Figs/riself.pdf}{1.0}}

\end{frame}


\begin{frame}[c]{Collaborative Cross}

  \figw{Figs/ri8.pdf}{1.0}

\end{frame}


\begin{frame}[c]{MAGIC}

  \figw{Figs/ri8self.pdf}{1.0}

\end{frame}



\begin{frame}[c]{Heterogeneous stock}

  \vspace{2mm}

  \figh{Figs/hs.pdf}{0.9}

\end{frame}


\begin{frame}[c]{Collaborative Cross}

  \figw{Figs/ri8.pdf}{1.0}

\end{frame}


\begin{frame}[c]{CC genome}

  \only<1>{\figw{Figs/ri8genome1.pdf}{1.0}}
  \only<2>{\figw{Figs/ri8genome2.pdf}{1.0}}
  \only<3>{\figw{Figs/ri8genome3.pdf}{1.0}}
  \only<4>{\figw{Figs/ri8genome4.pdf}{1.0}}
  \only<5>{\figw{Figs/ri8genome5.pdf}{1.0}}

\end{frame}


\begin{frame}[c]{Recombination fraction}
  \figw{Figs/basic_genetics.pdf}{1.0}
\end{frame}


\begin{frame}[c]{Simulation results}
  \figw{Figs/rf_by_sim.pdf}{1.0}
\end{frame}


\begin{frame}[c]{Haldane \& Waddington 1931}

  \figw{Figs/haldane_title.pdf}{0.9}

\end{frame}


\begin{frame}[c]{Result for selfing}

  \figw{Figs/haldane_selfing.png}{0.9}

\end{frame}


\begin{frame}[c]{Result for sib-mating}

  \only<1>{\figw{Figs/haldane_sibmating3.png}{0.9}}
  \only<2>{\figw{Figs/haldane_sibmating4.png}{0.9}}

\end{frame}


\begin{frame}[c]{Simulation results}
  \figw{Figs/rf_by_sim_genformula.pdf}{1.0}
\end{frame}



\begin{frame}[fragile]{Non-linear regression}

\vspace{5mm}

{
\verb|    out <- nls(| {\tt \vhilit R \verb|~| a*r/(1 + b*r)}\verb|,| \\
\verb|               data = data.frame(r=r, R=R),| \\
\verb|               start = list(a=4, b=6))| \\
\verb|    summary(out)|
}

\end{frame}


\begin{frame}[fragile]{Non-linear regression}
\addtocounter{framenumber}{-1}

\vspace{5mm}

\verb|    out <- nls(| {\tt \vhilit R \verb|~| a*r/(1 + b*r)}\verb|,| \\
\verb|               data = data.frame(r=r, R=R),| \\
\verb|               start = list(a=4, b=6))| \\
\verb|    summary(out)|

\vspace{8mm}

{\hilit
\verb|                          | \\
\verb|      Estimate  Std. Error| \\
\verb|    a    7.016       0.011| \\
\verb|    b    6.023       0.016|
}

\end{frame}


\begin{frame}[fragile]{Non-linear regression}
\addtocounter{framenumber}{-1}

\vspace{5mm}

\verb|    out <- nls(| {\tt \vhilit R \verb|~| a*r/(1 + b*r)}\verb|,| \\
\verb|               data = data.frame(r=r, R=R),| \\
\verb|               start = list(a=4, b=6))| \\
\verb|    summary(out)|

\vspace{8mm}

{\hilit
\verb|                                       More data      | \\
\verb|      Estimate  Std. Error        Estimate  Std. Error| \\
\verb|    a    7.016       0.011      a    7.003       0.008| \\
\verb|    b    6.023       0.016      b    6.005       0.012|
}

\end{frame}


\begin{frame}[c]{Simulation results}
  \figw{Figs/rf_by_sim_moredata.pdf}{1.0}
\end{frame}


\begin{frame}[c]{Markov chain}

\bbi
\item Sequence of random variables $\mathsf{\{X_0, X_1, X_2, \dots\}}$
  satisfying

\vspace{2mm}

\centerline{\hilit $\mathsf{Pr(X_{n+1} \ | \ X_0,
X_1, \dots, X_n) = Pr(X_{n+1} \ | \ X_n)}$}

\item Transition probabilities {\hilit $\mathsf{P_{ij} =
  Pr(X_{n+1} = j \ | \ X_n = i)}$}

\item Here, $\mathsf{X_n}$ = ``parental type'' at generation n.

\item We are interested in {\vhilit absorption probabilities}

\vspace{2mm}

\centerline{\hilit $\pi_j = \mathsf{Pr(X_n \rightarrow j \ | \ X_0)}$}
\ei

\end{frame}



\begin{frame}[c]{Absorption probabilities}

  \bbi

  \item[] Consider the case of {\vhilit absorption} into the state
$\left.\begin{array}{c} \text{A} \\ \text{A} \end{array} \right| \begin{array}{c} \text{A}
  \\ \text{A} \end{array} $

\hspace*{15mm} {\hilit (write this $\mathsf{AA|AA}$)}

\item[] Let $\mathsf{h_i}$ = probability, starting at i, of being absorbed into $\mathsf{AA|AA}$.

\item[] Then {\hilit $\mathsf{h_{AA|AA}}$ = 1} and
{\hilit $\mathsf{h_{AB|AB}}$ = 0}.

\item[] {\vhilit Condition on the first step:} \hspace*{10mm}
{\hilit
%h$_{\text{i}}$ = $\sum_{\text{k}}$ P$_{\text{ik}}$ h$_{\text{k}}$}
$\mathsf{h_i  = \sum_k  P_{ik} h_k}$}

\item[] For selfing, this gives a system of 3 linear equations.
\ei

\end{frame}



\begin{frame}[c]{Equations for selfing}

  \figw{Figs/haldane_selfing_equations.png}{0.9}

\end{frame}



\begin{frame}[c]{Equations for sib-mating}

  \figw{Figs/haldane_sibmating_equations.png}{0.9}

\end{frame}


\begin{frame}[c]{Result for sib-mating}

  \figw{Figs/haldane_sibmating3.png}{0.9}

\end{frame}



\begin{frame}[c]{3-point coincidence}

\hfill \includegraphics[scale=0.5]{Figs/threepoints.pdf}

\vspace*{-15mm}

  \bbi

\item $\mathsf{r_{ij}}$ = recombination fraction for interval (i, j)

{\hilit Assume $\mathsf{r_{12} = r_{23} = r}$.}

\item {\vhilit Coincidence} = $\mathsf{c = Pr(\text{double
  recombinant}) / r^2}$

\hspace{20mm} {\hilit = $\mathsf{Pr(\text{rec'n in 23} \ | \ \text{rec'n in 12}) /
  Pr(\text{rec'n in 23})}$}

\item
No interference { \hilit = 1 }

Positive interference { \hilit $<$ 1 }

Negative interference { \hilit $>$ 1  }


\item Generally {\hilit $\mathsf{c}$} is a function of {\hilit $\mathsf{r}$}

  \ei


\end{frame}



\begin{frame}[c]{Coincidence}
\only<1>{\figw{Figs/coincidence_ni.pdf}{1.0}}
\only<2>{\figw{Figs/coincidence_i.pdf}{1.0}}
\end{frame}



\begin{frame}[c]{Coincidence in 8-way RILs}


\bbi
\item The trick that allowed us to get the coincidence for 2-way RILs
  doesn't work for 8-way RILs.

\item It's sufficient to consider 4-way RILs.

\item Calculations for 3 points in 4-way RILs is still {\vhilit
  astoundingly complex}.


\bi
\item {\hilit 2 points in 2-way RILs by sib mating:}

55 parental types $\rightarrow$ {\vhilit 22 states} by symmetry

\item {\hilit 3 points in 4-way RILs by sib mating:}

2,164,240 parental types $\rightarrow$ {\vhilit 137,488 states}
by symmetry
\ei

\item Even {\vhilit counting} the states was difficult.
\ei
\end{frame}



\begin{frame}[c]{Coincidence}
\figw{Figs/coincidence_8way.pdf}{1.0}
\end{frame}




\begin{frame}[c]{The formula}

{ \fontsize{8.2pt}{9.2}\selectfont


$$ \mathsf{C = \frac{(1+6r)[280 + 1208r - 848r^2 + 5c(7-28r - 368r^2 + 344r^3)
  - 2c^2(49 - 324r + 452r^2)r^2 - 16c^3(1-2r)r^4]}{49 (1+12r-12cr^2)
    [5+10r-4(2+c)r^2+8cr^3]} }$$

}


\end{frame}



\begin{frame}[c]{3-point symmetry}

\centerline{\hspace*{15mm} \hilit
$\mathsf{Pr(M_2 = x \ | \ M_1 = A, M_2 \ne A, M_3 = A)}$
}
\figw{Figs/3pt_symmetry.pdf}{1.0}

\end{frame}


\begin{frame}[c]{Markov property}

\centerline{\hspace*{15mm} \hilit
$ \mathsf{log_2 \left\{ \frac{Pr(M_3 = A \ | \ M_2 = A, M_1 = x)}{Pr(M_3 = A \ | \ M_2 = A)} \right\} }$
}
\figw{Figs/3pt_markov1.pdf}{1.0}

\end{frame}

\begin{frame}[c]{Markov property}

\centerline{\hspace*{15mm} \hilit
$ \mathsf{log_2 \left\{ \frac{
    Pr(M_3 = A \ | \ M_2 = {\vhilit B}, M_1 = x)}{
    Pr(M_3 = A \ | \ M_2 = {\vhilit B})}\right\}}$
}
\figw{Figs/3pt_markov2.pdf}{1.0}

\end{frame}



\begin{frame}[c]{Markov property}

\centerline{\hspace*{15mm} \hilit
$ \mathsf{log_2 \left\{ \frac{
    Pr(M_3 = A \ | \ M_2 = {\vhilit C}, M_1 = x)}{
    Pr(M_3 = A \ | \ M_2 = {\vhilit C})}\right\}}$
}
\figw{Figs/3pt_markov3.pdf}{1.0}

\end{frame}


\begin{frame}[c]{Markov property}

\centerline{\hspace*{15mm} \hilit
$ \mathsf{log_2 \left\{ \frac{
    Pr(M_3 = A \ | \ M_2 = {\vhilit E}, M_1 = x)}{
    Pr(M_3 = A \ | \ M_2 = {\vhilit E})}\right\}}$
}
\figw{Figs/3pt_markov4.pdf}{1.0}

\end{frame}



\begin{frame}[c]{Whole genome simulations}

\bbi
\item 2-way selfing, 2-way sib-mating, 8-way sib-mating

\item Mouse-like genome, 1665 cM

\item Strong positive crossover interference

\item Inbreed to complex fixation

\item 10,000 simulation replicates
\ei

  \end{frame}


\begin{frame}[c]{No. generations to fixation}
\figw{Figs/ngen_fix.pdf}{1.0}
\end{frame}

\begin{frame}[c]{No. generations to 99\% fixation}
\figw{Figs/ngen99.pdf}{1.0}
\end{frame}

\begin{frame}[c]{Percent genome not fixed}
\figw{Figs/prophet.pdf}{1.0}
\end{frame}

\begin{frame}[c]{No. breakpoints}
\figw{Figs/nbrks.pdf}{1.0}
\end{frame}

\begin{frame}[c]{Segment lengths}
\figw{Figs/lseg_noarrows.pdf}{1.0}
\end{frame}

\begin{frame}[c]{Segment lengths}
\figw{Figs/lseg.pdf}{1.0}
\end{frame}

\begin{frame}[c]{Probability a segment is inherited intact}
\figw{Figs/probintact.pdf}{1.0}
\end{frame}

\begin{frame}[c]{Length of smallest segment}
\figw{Figs/lsmseg.pdf}{1.0}
\end{frame}

\begin{frame}[c]{No. segments $<$ 1 cM}
\figw{Figs/nsegsm1.pdf}{0.9}
\end{frame}



\begin{frame}[c]{Collaborative Cross}

  \figw{Figs/ri8.pdf}{1.0}

\end{frame}


\begin{frame}{The PreCC}


\vspace{5mm}

{\large What happens at $\mathsf{G_2 F_k}$?}

\bbi

\item[]  $\mathsf{Pr(g_1 = i)}$ \qquad\qquad {\hilit as a function of $\mathsf{k}$}

\item[]  $\mathsf{Pr(g_1 = i, g_2 = j)}$ \quad {\hilit as a function
      of $\mathsf{k}$ and the recombination fraction}
\ei

\end{frame}


\begin{frame}[c]{Crazy table}
\figw{Figs/preCC_hap_prob_table.pdf}{1.0}
\end{frame}


\begin{frame}[c]{Lesson}

  \centerline{\large Computer simulations are hugely valuable.}

\end{frame}



\begin{frame}[c]{Uses of simulations}

    \bbi
    \item Study probabilities
    \item Estimate power/sample size
    \item Evaluate performance of a method
    \item Evaluate sensitivity/robustness of a method
    \ei

\end{frame}



\begin{frame}[c]{Relative advantages?}

    \bbi
    \item Simulations
    \item Numerical calculations
    \item Analytic calculations
    \ei

\end{frame}




\begin{frame}[c]{References}

  \bbi

  \item Haldane \& Waddington (1931) Inbreeding and Linkage.
    16:357--374

  \item Broman KW (2005) The genomes of recombinant inbred lines. Genetics 169:1133--1146

  \item Teuscher \& Broman (2007) Haplotype probabilities for
    multiple-strain recombinant inbred lines. Genetics 175:1267--1274

  \item Broman KW (2012) Genotype probabilities at intermediate
    generations in the construction of recombinant inbred lines.
    Genetics 190:403--412

  \item Broman KW (2012) Haplotype probabilities in advanced
    intercross populations. G3 2:199--202

  \ei

\end{frame}

\end{document}
