\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}
\usepackage{eepic}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Model misspecification}
\subtitle{Estimating allele frequencies in sibships}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \note{}

} }


\begin{frame}[c]{Mapping disease genes}

  \centering
  \Large

  Look for genomic regions where \hspace*{20mm} \\[18pt]

  individuals with {\hilit similar phenotypes}

  also have {\hilit similar genotypes}

\end{frame}


\begin{frame}[c]{Affected sib pairs}
\only<1>{\figw{Figs/sibpairs.pdf}{1.0}}
\only<2>{\figw{Figs/sibpairs_wdata.pdf}{1.0}}
\end{frame}


\begin{frame}[c]{IBS vs IBD}


\large

IBS = identical by {\hilit state}

{\color{background} IBS}    = same allele number \\[18pt]

  IBD = identical by {\hilit descent}

{\color{background} IBD} = copies of the same ancestral allele \\[36pt]

non-inbred sibs are IBD = 0, 1, 2

{\color{background} non-inbred sibs} with probability = 1/4, 1/2, 1/4


\end{frame}

\begin{frame}[c]{Prostate cancer genome scan}
\figw{Figs/gh_results_bad.pdf}{1.0}
\end{frame}

\begin{frame}[c]{Lesson}

\centering
\Large
If it seems too good to be true, \\[12pt]
it probably is.

\end{frame}



\begin{frame}[c]{Prostate cancer pairs}
\only<1>{\figw{Figs/sibpairs_nopar.pdf}{1.0}}
\only<2>{\figw{Figs/sibpairs_nopar_wdata.pdf}{1.0}}
\end{frame}



\begin{frame}[c]{Prostate cancer genome scan -- corrected}
\figw{Figs/gh_results_good.pdf}{1.0}
\end{frame}


\begin{frame}{Estimating allele frequencies}

  \bigskip

  Usually, you would use the {\hilit founders} in the pedigrees. \\
  \qquad (assumed unrelated)

  \bigskip \bigskip

  What if you only have {\vhilit sibships}?

\end{frame}



\begin{frame}{Estimating allele frequencies with sibpairs}

  \bigskip

  {\hilit Method 1}: Use a random sibling from each

  \onslide<2->{

    $$ \text{var}(\hat{p}^{(1)}) = \frac{p(1-p)}{2n}$$

  }


  \bigskip \bigskip \bigskip

  {\hilit Method 2}: Use everyone, ignoring relationships


  \onslide<2->{

    $$ \text{var}(\hat{p}^{(2)}) = \only<2>{\, \text{\bf ?}}
    \only<3->{{\vhilit (3/4)} \, \left(\frac{p(1-p)}{2n}\right)}
    $$
  }

  \only<4->{\qquad relative efficiency = 4/3 = 1.33

    \qquad \qquad {\lolit (best possible = 1.5)}}

\end{frame}



\begin{frame}[c]{My favorite equations}
  \begin{eqnarray*}
    \text{E}(X) & = & \text{E}[ \text{E}(X|Z) ] \\[18pt]
    \text{var}(X) & = & \text{E}[ \text{var}(X|Z) ] + \text{var}[ \text{E}(X|Z) ] \\[18pt]
    \text{cov}(X,Y) & = & \text{E}[ \text{cov}(X,Y|Z) ] + \text{cov}[ \text{E}(X|Z), \text{E}(Y|Z) ]
  \end{eqnarray*}

  \bigskip\bigskip

  \hfill {\lolit Everything is a mixture}

\end{frame}


\begin{frame}[c]{Another fave}
  \begin{eqnarray*}
    \text{cov}(X, aY+bZ) & = & a \, \text{cov}(X,Y) + b \, \text{cov}(X,Z) \\[18pt]
    \text{{\lolit thus} \, var}(X+Y) & = & \text{cov}(X+Y,X+Y) \\
    & = & \text{cov}(X+Y, X) + \text{cov}(X+Y, Y) \\
    & = & \text{cov}(X, X) + \text{cov}(X, Y) + \text{cov}(Y,X) + \text{cov}(Y,Y) \\
    & = & \text{var}(X) + \text{var}(Y) + 2 \, \text{cov}(X,Y)
    \end{eqnarray*}
\end{frame}


\begin{frame}[c]{}
  \figh{Figs/mixture_univariate.pdf}{1.0}
\end{frame}

\begin{frame}[c]{}
  \figh{Figs/mixture_bivariate.pdf}{1.0}
\end{frame}








\begin{frame}{Average relative efficiency}
\figh{Figs/ave_rel_eff.png}{0.7}

\vspace{10mm}

\hfill \footnotesize {\lolit Broman (2001)
  \href{https://doi.org/10.1002/gepi.2}{doi:10.1002/gepi.2}}

\end{frame}

\begin{frame}{Method 3}
\figh{Figs/rel_eff_method3.png}{0.6}

\vspace{18mm}

\hfill \footnotesize {\lolit Broman (2001)
  \href{https://doi.org/10.1002/gepi.2}{doi:10.1002/gepi.2}}

\end{frame}

\begin{frame}{Method 4}
\figh{Figs/rel_eff_method4.png}{0.75}

\bigskip

\hfill \footnotesize {\lolit Broman (2001)
  \href{https://doi.org/10.1002/gepi.2}{doi:10.1002/gepi.2}}

\end{frame}



\end{document}
