\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{../LaTeX/header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Maintaining, supporting, and sustaining \\ scientific software}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Course web: \href{https://kbroman.org/AdvData}{\tt kbroman.org/AdvData}}
}

\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \note{
    This lecture is about developing and maintaining scientific
    software, and the value of this endeavor for academic data
    scientists.
}
} }



\begin{frame}[c]{20 years of R/qtl}

\figw{Figs/rqtl_lines_code.pdf}{1.0}

\note{
   For more than 20 years, I've been developing an R package for QTL
   mapping, R/qtl. This graph shows the number of lines of R code, C
   code, and documentation in each release of the package.

   If I'd known what I was getting into, I maybe wouldn't have started
   the project. But it's actually been hugely valuable for me. I've
   made a lot of friends and formed a lot of great collaborations from
   the work. And it's been useful to the community, as well.

   Today I want to talk about this experience.
}
\end{frame}





\begin{frame}{}

\vspace*{16.7mm}

\centerline{\Large Why?}

\note{
  I've put a lot of effort into this software, and not just in
  developing it, but also in providing support to users. Why do so?

  First, the software is important for my own research work, and
  particularly for many of my collaborative research, and I've come to
  learn that improvements aimed at making the software simpler to use
  have also made it simpler for {\hilit me} to use.

  Second, if you want people to use your software, you need to help
  them get started. And not just in providing them tutorials and other
  documentation, but also in that first nasty step of getting their
  data into the right form. If they get frustrated at that first step,
  they'll just give up and look elsewhere.

  Third, I've formed a lot of new collaborations through the
  software. People use it but find that their own data present
  difficulties that they can't quite figure out, and they ask me for
  help. Some of that short-term helps turns into longer
  collaborations.

  Finally, the software has been a great platform for me to provide my
  own methods developments. If I want people to use my methods ideas,
  I need to provide them software; software integrated into a
  widely-used tool will be easier to pick up.
}
\end{frame}



\begin{frame}[c]{}

\centerline{\Large Good things}

\vspace{4mm}

\onslide<2>{
\begin{itemize}
\lolit
  \item some of the code
  \item basics of the user interface
  \item diagnostics and data visualization
  \item quite comprehensive
  \item quite flexible
\end{itemize}

}

\note{
  When you've been working on a project for a long time, you start to
  focus primarily on its weaknesses, and it can be difficult to
  identify any good aspects anymore.

  But there are a number of good things about R/qtl. Some of the code,
  particularly the HMM for handling missing genotype information, is
  quite good. And the basic user interface was rather a nice idea; it
  hides the complexities while still making them accessible to an
  advanced user.

  And the data diagnostics and data visualizations are quite good. And
  overall the software is both quite comprehensive and quite flexible.
  It has served me well, overall.
}
\end{frame}


\begin{frame}{}

\vspace*{16.7mm}

\centerline{\Large Bad things}

\note{
  Bad things are more easy to identify. Really, it's hard not to just
  see all of the bad things.

  Some of the code is really terrible and complicated. Short-term
  efforts to fix problems were accomplished via band-aids that made
  the code more complicated and confusing and harder to maintain.

  And there was never any defined specifications on the data format or
  describing the software design, which makes it much harder for
  others to get involved.

  Small changes in one place have implications throughout the code
  base that are to identify, making bug fixes or further developments
  much more difficult.
}
\end{frame}


\begin{frame}[c]{Input file}

\figh{Figs/datafile.pdf}{0.6}

\note{
  To illustrate one of the bad things, let's first look at the data
  input format. A single comma-delimited file contains the three
  aspects of the data: the phenotypes, the genotypes, and the marker
  data.

  Each row is an individual, the initial columns are the phenotypes,
  and the subsequent columns are the marker genotypes. For the
  markers, the second row indicates the chromosome assignments, and the
  third row contains the locations. The initial cells in the second
  and third rows need to be completely blank.

  The first thing, upon reading in the data, is to split the data into
  the three parts: the phenotypes from the genotypes and then the
  marker map from the genotypes. We do that first by identifying the
  first non-blank cell in the second row.

  When R/qtl started to be used for larger data sets (such as with gene
  expression phenotypes, say on 30,000 genes), users complained about
  how long it took to load a file. I'd ask ``Well, how much data do
  you have?'' and then say ``Well, that's a big file, it's going to
  take a while.''

  But then I had a data set with like 1500 traits and 200 mice and it
  took a full minute to load, and that just seemed wrong, and so I
  finally looked to see what was happening.
}
\end{frame}


\begin{frame}[c,fragile]{Stupidest code ever}

\begin{center}
\begin{minipage}[c]{9.3cm}
\begin{semiverbatim}
\lstset{basicstyle=\normalsize}
\begin{lstlisting}[linewidth=9.3cm]
n <- ncol(data)
temp <- rep(FALSE,n)
for(i in 1:n) {
  temp[i] <- all(data[2,1:i]=="")
  if(!temp[i]) break
}
if(!any(temp)) stop("...")
n.phe <- max((1:n)[temp])
\end{lstlisting}
\end{semiverbatim}
\end{minipage}
\end{center}

\vspace{3mm}

\hfill \href{http://kbroman.org/blog/2011/08/17/the-stupidest-r-code-ever}{\scriptsize \lolit \tt kbroman.org/blog/2011/08/17/the-stupidest-r-code-ever}

\note{
  And this is what I found. In the {\tt for} loop, I'm look at the
  first cell, and then the first two cells, then the first three
  cells, etc., until I get to a point where they are not all empty.
  Then I brake out of the loop.

  It turned out that for my file with 1500 traits and 200 mice, it
  took like 2 seconds to read the data but 58 seconds to find this
  spot to split the phenotypes and genotypes.

  This code is no longer in R/qtl, but it's embarrassing to see how
  long it {\hilit was} part of the package.
}
\end{frame}


\begin{frame}[c]{}

  \large

  {\hilit Open source} {\lolit means}

  everyone can see my stupid mistakes

  \bigskip \bigskip \bigskip

  \onslide<2>{
    {\hilit Version control} {\lolit means}

  everyone can see every stupid mistake I've ever made
}

\note{
  This is one of my favorite things to say.
}
\end{frame}




\begin{frame}[c]{More typically bad code}

  \large
      The {\hilit \tt scantwo()} function is {\hilit 1446 lines} long.

      \bigskip \bigskip

      The related C code is 20\% of the C code in R/qtl.


\note{
  This sort of thing really needs to be broken up into smaller pieces.

  As it is, it's really hard to extend and maintain.
}
\end{frame}



\begin{frame}[c]{Baroque data structures}

  \large
      {\tt attr(mycross\$geno[["X"]]\$probs, "map")}


\note{
  The data structures in R/qtl really got out of control: far too
  deeply nested.

  Attributes can be great, but when I learned about them, I started to
  pile all sorts of shit in there.
}
\end{frame}






\begin{frame}{}

\vspace*{16.7mm}

\centerline{\Large Documentation}

\note{
   I've learned a lot in the last 20 years. First, regarding
   documentation. I've written a lot of detailed documentation. But
   most of it is not at all read. All those help files carefully
   describing the use of each function? They're mostly not read. Users
   do like the examples, but the rest is generally too technical
   really just useful as a reference rather than for learning or
   trouble-shooting.

   What folks really want are very tailored tutorials: basically case
   studies showing how to use the software and how to interpret the
   results, ideally with data and with analysis goals that are very
   close to the users' applications.

   So by all means write the detailed documentation, but focus most of
   your effort on writing vignettes or other tutorials that match what
   your users are trying to do with the software.
}
\end{frame}



\begin{frame}{}

\vspace*{16.7mm}

  \only<1>{\centerline{\Large User support}}

  \only<2|handout 0>{
    \centerline{\Large ``I tried X and it didn't work.''}}

  \only<3|handout 0>{
    \large ``Could you look at the attached 25-page Word
      document containing code and output and tell me if I'm doing
      something wrong?''}


\note{
   Supporting the software, by answering users' questions, is time
   consuming and can be frustrating (both for me and for the users),
   but it is where you can have the greatest impact. Your responses to
   questions can be the difference between users continuing with the
   software and giving up in frustration.

   Many times users don't really seem to understand how to ask
   questions. I generally want to some details about the nature of the
   data, and then the exact code and output and error messages.

   I often get questions like, ``Can you look at the attached 20 page
   Word document and let me know if I'm doing this okay?'' Or, ``I
   tried X and it didn't work.''
   I've found that it's best to take a short break between when I read
   a question and when I respond. Often I'll be really annoyed by a
   question initially, but if I wait 30 minutes, I can return to it
   and respond in a much more positive way.

   I try to focus on my own frustrating experiences with software, and
   recall that while for me it may be the 100th time I've heard a
   particular question, for the user it's the first time.

    I'd hope to build a ``community'' of people answering each other's
    questions, but it continues to be largely me doing the answering.
}
\end{frame}


\begin{frame}{}

\vspace*{16.7mm}

  \centerline{\Large Incorporating others' code}

  \note{
    It's great to get contributions from others. But once you've incorporated
    their features, you're responsible for maintaining and supporting
    it.

    I now think that big things should mostly be made to be separate
    packages.
  }
\end{frame}



\begin{frame}{}

\vspace*{16.7mm}

  \centerline{\Large Version control}

  \note{
    How on earth did I get by before git?

    Really critical for incorporating others' changes, and for trying out
    new things without breaking what's working.
  }
\end{frame}

\begin{frame}{}

\vspace*{16.7mm}

  \centerline{\Large Tests}

  \note{
    How on earth does any of this work? There are basically no tests
    that the code works.

    I'm now thoroughly in love with unit tests and
    Hadley's testthat.
  }
\end{frame}





\begin{frame}[c]{QTL mapping}

\vspace{5mm}
\figw{Figs/lodcurve_insulin_with_effects.pdf}{1.1}

\note{
}
\end{frame}





\begin{frame}[c]{Heterogeneous stock}

  \vspace{2mm}

  \figh{Figs/hs.pdf}{0.9}

\note{
}
\end{frame}



\begin{frame}[c]{Challenges: {\color{foreground} scale of results}}

\only<1|handout 0>{\figw{Figs/scale_fig1.pdf}{1.0}}
\only<2>{\figw{Figs/scale_fig2.pdf}{1.0}}

\note{
}
\end{frame}




\begin{frame}[c]{Challenges: {\color{foreground} organizing, automating}}

\only<1|handout 0>{\figw{Figs/batches_fig1.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/batches_fig2.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/batches_fig3.pdf}{1.0}}
\only<4|handout 0>{\figw{Figs/batches_fig4.pdf}{1.0}}
\only<5|handout 0>{\figw{Figs/batches_fig5.pdf}{1.0}}
\only<6|handout 0>{\figw{Figs/batches_fig6.pdf}{1.0}}
\only<7>{\figw{Figs/batches_fig7.pdf}{1.0}}

\note{
}
\end{frame}



\begin{frame}[c]{}

  \vspace*{5mm}

\figh{Figs/rqtl2_3d.png}{0.85}


\vspace{3mm}

\hfill \href{https://ropenscilabs.github.io/miner_book}{\scriptsize \lolit \tt ropenscilabs.github.io/miner\_book}

\note{
}
\end{frame}



\begin{frame}[c]{R/qtl2}

\vspace*{-16.2mm}

  \vspace{21mm}

  \bbi
\item High-density genotypes
\item High-dimensional phenotypes
\item Multi-parent populations
\item Linear mixed models
  \ei

  \vspace{25mm}

\hfill \href{http://kbroman.org/qtl2}{\small \tt kbroman.org/qtl2}

\note{
}
\end{frame}



\begin{frame}{R/qtl2: \color{foreground} Let's not make the same mistakes}

  \bbi
\only<1>{
\item C++ and Rcpp
\item Roxygen2 for documentation
\item Unit tests
\item A single ``switch'' for cross type
}
\only<2|handout 0>{
{\lolit
\item C++ and Rcpp
\item Roxygen2 for documentation
\item Unit tests
\item A single ``switch'' for cross type
}
}
\onslide<2>{
\item Yet another data input format
\item Flatter data structures, but still complex
}
\ei

\note{
}
\end{frame}



\begin{frame}{}

\vspace*{16.7mm}

\centerline{\Large Sustainable academic software}

\note{
}
\end{frame}







\begin{frame}[c]{Acknowledgments}

\begin{columns}[T]
  \begin{column}[T]{0.5\textwidth}
    \vspace{0pt}
\bi
\item[] Danny Arends
\item[] Gary Churchill
\item[] Nick Furlotte
\item[] Dan Gatti
\item[] Ritsert Jansen
\item[] Pjotr Prins
\item[] \'Saunak Sen
\item[] Petr Simecek
\item[] Artem Tarasov
\item[] Hao Wu
\item[] Brian Yandell
  \ei
  \end{column} \hfill
\begin{column}[T]{0.5\textwidth}
\vspace*{0mm}

  \bi
\item[] Robert Corty
\item[] Timoth\'ee Flutre
\item[] Lars Ronnegard
\item[] Rohan Shah
\item[] Laura Shannon
\item[] Quoc Tran
\item[] Aaron Wolen
\item[]
\item[] NIH/NIGMS
  \ei
\end{column}
\end{columns}

\note{
}
\end{frame}

\end{document}
